// @aws-sdk/client-rekognition/dist/cjs/commands/CreateProjectVersionCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateProjectVersionCommand = void 0;
const index_1 = require("../models/index");
const Aws_json1_1_1 = require("../protocols/Aws_json1_1");
const middleware_serde_1 = require("@aws-sdk/middleware-serde");
const smithy_client_1 = require("@aws-sdk/smithy-client");
class CreateProjectVersionCommand extends smithy_client_1.Command {
    // Start section: command_properties
    // End section: command_properties
    constructor(input) {
        // Start section: command_constructor
        super();
        this.input = input;
        // End section: command_constructor
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(middleware_serde_1.getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const handlerExecutionContext = {
            logger,
            inputFilterSensitiveLog: index_1.CreateProjectVersionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: index_1.CreateProjectVersionResponse.filterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return Aws_json1_1_1.serializeAws_json1_1CreateProjectVersionCommand(input, context);
    }
    deserialize(output, context) {
        return Aws_json1_1_1.deserializeAws_json1_1CreateProjectVersionCommand(output, context);
    }
}
exports.CreateProjectVersionCommand = CreateProjectVersionCommand;
