// @aws-sdk/client-s3/dist/cjs/models/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebsiteConfiguration = exports.VersioningConfiguration = exports.UploadPartRequest = exports.UploadPartOutput = exports.UploadPartCopyRequest = exports.UploadPartCopyOutput = exports.Transition = exports.TopicConfiguration = exports.TargetGrant = exports.Tagging = exports.Tag = exports.StorageClassAnalysisDataExport = exports.StorageClassAnalysis = exports.StatsEvent = exports.Stats = exports.SSES3 = exports.SseKmsEncryptedObjects = exports.SSEKMS = exports.SourceSelectionCriteria = exports.ServerSideEncryptionRule = exports.ServerSideEncryptionConfiguration = exports.ServerSideEncryptionByDefault = exports.SelectParameters = exports.SelectObjectContentRequest = exports.SelectObjectContentOutput = exports.SelectObjectContentEventStream = exports.ScanRange = exports.S3Location = exports.S3KeyFilter = exports.RoutingRule = exports.RestoreRequestType = exports.RestoreRequest = exports.RestoreObjectRequest = exports.RestoreObjectOutput = exports.RequestProgress = exports.RequestPaymentConfiguration = exports.ReplicationTimeValue = exports.ReplicationTime = exports.ReplicationRuleFilter = exports.ReplicationRuleAndOperator = exports.ReplicationRule = exports.ReplicationConfiguration = exports.RedirectAllRequestsTo = exports.Redirect = exports.RecordsEvent = exports.QuoteFields = exports.QueueConfiguration = exports.PutPublicAccessBlockRequest = exports.PutObjectTaggingRequest = exports.PutObjectTaggingOutput = exports.PutObjectRetentionRequest = exports.PutObjectRetentionOutput = exports.PutObjectRequest = exports.PutObjectOutput = exports.PutObjectLockConfigurationRequest = exports.PutObjectLockConfigurationOutput = exports.PutObjectLegalHoldRequest = exports.PutObjectLegalHoldOutput = exports.PutObjectAclRequest = exports.PutObjectAclOutput = exports.PutBucketWebsiteRequest = exports.PutBucketVersioningRequest = exports.PutBucketTaggingRequest = exports.PutBucketRequestPaymentRequest = exports.PutBucketReplicationRequest = exports.PutBucketPolicyRequest = exports.PutBucketNotificationConfigurationRequest = exports.PutBucketMetricsConfigurationRequest = exports.PutBucketLoggingRequest = exports.PutBucketLifecycleConfigurationRequest = exports.PutBucketInventoryConfigurationRequest = exports.PutBucketEncryptionRequest = exports.PutBucketCorsRequest = exports.PutBucketAnalyticsConfigurationRequest = exports.PutBucketAclRequest = exports.PutBucketAccelerateConfigurationRequest = exports.PublicAccessBlockConfiguration = exports.ProgressEvent = exports.Progress = exports.PolicyStatus = exports.Part = exports.ParquetInput = exports.Owner = exports.OutputSerialization = exports.OutputLocation = exports.ObjectVersion = exports.ObjectNotInActiveTierError = exports.ObjectLockRule = exports.ObjectLockRetention = exports.ObjectLockLegalHold = exports.ObjectLockConfiguration = exports.ObjectIdentifier = exports.ObjectAlreadyInActiveTierError = exports._Object = exports.NotificationConfigurationFilter = exports.NotificationConfiguration = exports.NoSuchUpload = exports.NoSuchKey = exports.NoSuchBucket = exports.NoncurrentVersionTransition = exports.NoncurrentVersionExpiration = exports.MultipartUpload = exports.MetricsFilter = exports.MetricsConfiguration = exports.MetricsAndOperator = exports.Metrics = exports.MetadataEntry = exports.LoggingEnabled = exports.ListPartsRequest = exports.ListPartsOutput = exports.ListObjectVersionsRequest = exports.ListObjectVersionsOutput = exports.ListObjectsV2Request = exports.ListObjectsV2Output = exports.ListObjectsRequest = exports.ListObjectsOutput = exports.ListMultipartUploadsRequest = exports.ListMultipartUploadsOutput = exports.ListBucketsOutput = exports.ListBucketMetricsConfigurationsRequest = exports.ListBucketMetricsConfigurationsOutput = exports.ListBucketInventoryConfigurationsRequest = exports.ListBucketInventoryConfigurationsOutput = exports.ListBucketAnalyticsConfigurationsRequest = exports.ListBucketAnalyticsConfigurationsOutput = exports.LifecycleRuleFilter = exports.LifecycleRuleAndOperator = exports.LifecycleRule = exports.LifecycleExpiration = exports.LambdaFunctionConfiguration = exports.JSONType = exports.JSONOutput = exports.JSONInput = exports.InventorySchedule = exports.InventoryS3BucketDestination = exports.InventoryFilter = exports.InventoryEncryption = exports.InventoryDestination = exports.InventoryConfiguration = exports.InputSerialization = exports.Initiator = exports.IndexDocument = exports.HeadObjectRequest = exports.HeadObjectOutput = exports.HeadBucketRequest = exports.Grantee = exports.Grant = exports.GlacierJobParameters = exports.GetPublicAccessBlockRequest = exports.GetPublicAccessBlockOutput = exports.GetObjectTorrentRequest = exports.GetObjectTorrentOutput = exports.GetObjectTaggingRequest = exports.GetObjectTaggingOutput = exports.GetObjectRetentionRequest = exports.GetObjectRetentionOutput = exports.GetObjectRequest = exports.GetObjectOutput = exports.GetObjectLockConfigurationRequest = exports.GetObjectLockConfigurationOutput = exports.GetObjectLegalHoldRequest = exports.GetObjectLegalHoldOutput = exports.GetObjectAclRequest = exports.GetObjectAclOutput = exports.GetBucketWebsiteRequest = exports.GetBucketWebsiteOutput = exports.GetBucketVersioningRequest = exports.GetBucketVersioningOutput = exports.GetBucketTaggingRequest = exports.GetBucketTaggingOutput = exports.GetBucketRequestPaymentRequest = exports.GetBucketRequestPaymentOutput = exports.GetBucketReplicationRequest = exports.GetBucketReplicationOutput = exports.GetBucketPolicyStatusRequest = exports.GetBucketPolicyStatusOutput = exports.GetBucketPolicyRequest = exports.GetBucketPolicyOutput = exports.GetBucketNotificationConfigurationRequest = exports.GetBucketMetricsConfigurationRequest = exports.GetBucketMetricsConfigurationOutput = exports.GetBucketLoggingRequest = exports.GetBucketLoggingOutput = exports.GetBucketLocationRequest = exports.GetBucketLocationOutput = exports.GetBucketLifecycleConfigurationRequest = exports.GetBucketLifecycleConfigurationOutput = exports.GetBucketInventoryConfigurationRequest = exports.GetBucketInventoryConfigurationOutput = exports.GetBucketEncryptionRequest = exports.GetBucketEncryptionOutput = exports.GetBucketCorsRequest = exports.GetBucketCorsOutput = exports.GetBucketAnalyticsConfigurationRequest = exports.GetBucketAnalyticsConfigurationOutput = exports.GetBucketAclRequest = exports.GetBucketAclOutput = exports.GetBucketAccelerateConfigurationRequest = exports.GetBucketAccelerateConfigurationOutput = exports.FilterRule = exports.FileHeaderInfo = exports.ExistingObjectReplication = exports.ErrorDocument = exports._Error = exports.EndEvent = exports.EncryptionConfiguration = exports.Encryption = exports.Destination = exports.DeletePublicAccessBlockRequest = exports.DeleteObjectTaggingRequest = exports.DeleteObjectTaggingOutput = exports.DeleteObjectsRequest = exports.DeleteObjectsOutput = exports.DeleteObjectRequest = exports.DeleteObjectOutput = exports.DeleteMarkerReplication = exports.DeleteMarkerEntry = exports.DeletedObject = exports.DeleteBucketWebsiteRequest = exports.DeleteBucketTaggingRequest = exports.DeleteBucketRequest = exports.DeleteBucketReplicationRequest = exports.DeleteBucketPolicyRequest = exports.DeleteBucketMetricsConfigurationRequest = exports.DeleteBucketLifecycleRequest = exports.DeleteBucketInventoryConfigurationRequest = exports.DeleteBucketEncryptionRequest = exports.DeleteBucketCorsRequest = exports.DeleteBucketAnalyticsConfigurationRequest = exports.Delete = exports.DefaultRetention = exports.CSVOutput = exports.CSVInput = exports.CreateMultipartUploadRequest = exports.CreateMultipartUploadOutput = exports.CreateBucketRequest = exports.CreateBucketOutput = exports.CreateBucketConfiguration = exports.CORSRule = exports.CORSConfiguration = exports.CopyPartResult = exports.CopyObjectResult = exports.CopyObjectRequest = exports.CopyObjectOutput = exports.ContinuationEvent = exports.Condition = exports.CompleteMultipartUploadRequest = exports.CompleteMultipartUploadOutput = exports.CompletedPart = exports.CompletedMultipartUpload = exports.CommonPrefix = exports.BucketLoggingStatus = exports.BucketLifecycleConfiguration = exports.BucketAlreadyOwnedByYou = exports.BucketAlreadyExists = exports.Bucket = exports.AnalyticsS3BucketDestination = exports.AnalyticsFilter = exports.AnalyticsExportDestination = exports.AnalyticsConfiguration = exports.AnalyticsAndOperator = exports.AccessControlTranslation = exports.AccessControlPolicy = exports.AccelerateConfiguration = exports.AbortMultipartUploadRequest = exports.AbortMultipartUploadOutput = exports.AbortIncompleteMultipartUpload = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
var AbortIncompleteMultipartUpload;
(function (AbortIncompleteMultipartUpload) {
    AbortIncompleteMultipartUpload.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    AbortIncompleteMultipartUpload.isa = (o) => smithy_client_1.isa(o, "AbortIncompleteMultipartUpload");
})(AbortIncompleteMultipartUpload = exports.AbortIncompleteMultipartUpload || (exports.AbortIncompleteMultipartUpload = {}));
var AbortMultipartUploadOutput;
(function (AbortMultipartUploadOutput) {
    AbortMultipartUploadOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    AbortMultipartUploadOutput.isa = (o) => smithy_client_1.isa(o, "AbortMultipartUploadOutput");
})(AbortMultipartUploadOutput = exports.AbortMultipartUploadOutput || (exports.AbortMultipartUploadOutput = {}));
var AbortMultipartUploadRequest;
(function (AbortMultipartUploadRequest) {
    AbortMultipartUploadRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    AbortMultipartUploadRequest.isa = (o) => smithy_client_1.isa(o, "AbortMultipartUploadRequest");
})(AbortMultipartUploadRequest = exports.AbortMultipartUploadRequest || (exports.AbortMultipartUploadRequest = {}));
var AccelerateConfiguration;
(function (AccelerateConfiguration) {
    AccelerateConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    AccelerateConfiguration.isa = (o) => smithy_client_1.isa(o, "AccelerateConfiguration");
})(AccelerateConfiguration = exports.AccelerateConfiguration || (exports.AccelerateConfiguration = {}));
var AccessControlPolicy;
(function (AccessControlPolicy) {
    AccessControlPolicy.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    AccessControlPolicy.isa = (o) => smithy_client_1.isa(o, "AccessControlPolicy");
})(AccessControlPolicy = exports.AccessControlPolicy || (exports.AccessControlPolicy = {}));
var AccessControlTranslation;
(function (AccessControlTranslation) {
    AccessControlTranslation.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    AccessControlTranslation.isa = (o) => smithy_client_1.isa(o, "AccessControlTranslation");
})(AccessControlTranslation = exports.AccessControlTranslation || (exports.AccessControlTranslation = {}));
var AnalyticsAndOperator;
(function (AnalyticsAndOperator) {
    AnalyticsAndOperator.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    AnalyticsAndOperator.isa = (o) => smithy_client_1.isa(o, "AnalyticsAndOperator");
})(AnalyticsAndOperator = exports.AnalyticsAndOperator || (exports.AnalyticsAndOperator = {}));
var AnalyticsConfiguration;
(function (AnalyticsConfiguration) {
    AnalyticsConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    AnalyticsConfiguration.isa = (o) => smithy_client_1.isa(o, "AnalyticsConfiguration");
})(AnalyticsConfiguration = exports.AnalyticsConfiguration || (exports.AnalyticsConfiguration = {}));
var AnalyticsExportDestination;
(function (AnalyticsExportDestination) {
    AnalyticsExportDestination.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    AnalyticsExportDestination.isa = (o) => smithy_client_1.isa(o, "AnalyticsExportDestination");
})(AnalyticsExportDestination = exports.AnalyticsExportDestination || (exports.AnalyticsExportDestination = {}));
var AnalyticsFilter;
(function (AnalyticsFilter) {
    AnalyticsFilter.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    AnalyticsFilter.isa = (o) => smithy_client_1.isa(o, "AnalyticsFilter");
})(AnalyticsFilter = exports.AnalyticsFilter || (exports.AnalyticsFilter = {}));
var AnalyticsS3BucketDestination;
(function (AnalyticsS3BucketDestination) {
    AnalyticsS3BucketDestination.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    AnalyticsS3BucketDestination.isa = (o) => smithy_client_1.isa(o, "AnalyticsS3BucketDestination");
})(AnalyticsS3BucketDestination = exports.AnalyticsS3BucketDestination || (exports.AnalyticsS3BucketDestination = {}));
var Bucket;
(function (Bucket) {
    Bucket.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Bucket.isa = (o) => smithy_client_1.isa(o, "Bucket");
})(Bucket = exports.Bucket || (exports.Bucket = {}));
var BucketAlreadyExists;
(function (BucketAlreadyExists) {
    BucketAlreadyExists.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    BucketAlreadyExists.isa = (o) => smithy_client_1.isa(o, "BucketAlreadyExists");
})(BucketAlreadyExists = exports.BucketAlreadyExists || (exports.BucketAlreadyExists = {}));
var BucketAlreadyOwnedByYou;
(function (BucketAlreadyOwnedByYou) {
    BucketAlreadyOwnedByYou.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    BucketAlreadyOwnedByYou.isa = (o) => smithy_client_1.isa(o, "BucketAlreadyOwnedByYou");
})(BucketAlreadyOwnedByYou = exports.BucketAlreadyOwnedByYou || (exports.BucketAlreadyOwnedByYou = {}));
var BucketLifecycleConfiguration;
(function (BucketLifecycleConfiguration) {
    BucketLifecycleConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    BucketLifecycleConfiguration.isa = (o) => smithy_client_1.isa(o, "BucketLifecycleConfiguration");
})(BucketLifecycleConfiguration = exports.BucketLifecycleConfiguration || (exports.BucketLifecycleConfiguration = {}));
var BucketLoggingStatus;
(function (BucketLoggingStatus) {
    BucketLoggingStatus.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    BucketLoggingStatus.isa = (o) => smithy_client_1.isa(o, "BucketLoggingStatus");
})(BucketLoggingStatus = exports.BucketLoggingStatus || (exports.BucketLoggingStatus = {}));
var CommonPrefix;
(function (CommonPrefix) {
    CommonPrefix.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    CommonPrefix.isa = (o) => smithy_client_1.isa(o, "CommonPrefix");
})(CommonPrefix = exports.CommonPrefix || (exports.CommonPrefix = {}));
var CompletedMultipartUpload;
(function (CompletedMultipartUpload) {
    CompletedMultipartUpload.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    CompletedMultipartUpload.isa = (o) => smithy_client_1.isa(o, "CompletedMultipartUpload");
})(CompletedMultipartUpload = exports.CompletedMultipartUpload || (exports.CompletedMultipartUpload = {}));
var CompletedPart;
(function (CompletedPart) {
    CompletedPart.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    CompletedPart.isa = (o) => smithy_client_1.isa(o, "CompletedPart");
})(CompletedPart = exports.CompletedPart || (exports.CompletedPart = {}));
var CompleteMultipartUploadOutput;
(function (CompleteMultipartUploadOutput) {
    CompleteMultipartUploadOutput.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.SSEKMSKeyId && { SSEKMSKeyId: smithy_client_1.SENSITIVE_STRING })));
    CompleteMultipartUploadOutput.isa = (o) => smithy_client_1.isa(o, "CompleteMultipartUploadOutput");
})(CompleteMultipartUploadOutput = exports.CompleteMultipartUploadOutput || (exports.CompleteMultipartUploadOutput = {}));
var CompleteMultipartUploadRequest;
(function (CompleteMultipartUploadRequest) {
    CompleteMultipartUploadRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    CompleteMultipartUploadRequest.isa = (o) => smithy_client_1.isa(o, "CompleteMultipartUploadRequest");
})(CompleteMultipartUploadRequest = exports.CompleteMultipartUploadRequest || (exports.CompleteMultipartUploadRequest = {}));
var Condition;
(function (Condition) {
    Condition.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Condition.isa = (o) => smithy_client_1.isa(o, "Condition");
})(Condition = exports.Condition || (exports.Condition = {}));
var ContinuationEvent;
(function (ContinuationEvent) {
    ContinuationEvent.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ContinuationEvent.isa = (o) => smithy_client_1.isa(o, "ContinuationEvent");
})(ContinuationEvent = exports.ContinuationEvent || (exports.ContinuationEvent = {}));
var CopyObjectOutput;
(function (CopyObjectOutput) {
    CopyObjectOutput.filterSensitiveLog = (obj) => (Object.assign(Object.assign(Object.assign({}, obj), (obj.SSEKMSKeyId && { SSEKMSKeyId: smithy_client_1.SENSITIVE_STRING })), (obj.SSEKMSEncryptionContext && { SSEKMSEncryptionContext: smithy_client_1.SENSITIVE_STRING })));
    CopyObjectOutput.isa = (o) => smithy_client_1.isa(o, "CopyObjectOutput");
})(CopyObjectOutput = exports.CopyObjectOutput || (exports.CopyObjectOutput = {}));
var CopyObjectRequest;
(function (CopyObjectRequest) {
    CopyObjectRequest.filterSensitiveLog = (obj) => (Object.assign(Object.assign(Object.assign(Object.assign(Object.assign({}, obj), (obj.SSEKMSKeyId && { SSEKMSKeyId: smithy_client_1.SENSITIVE_STRING })), (obj.SSECustomerKey && { SSECustomerKey: smithy_client_1.SENSITIVE_STRING })), (obj.CopySourceSSECustomerKey && { CopySourceSSECustomerKey: smithy_client_1.SENSITIVE_STRING })), (obj.SSEKMSEncryptionContext && { SSEKMSEncryptionContext: smithy_client_1.SENSITIVE_STRING })));
    CopyObjectRequest.isa = (o) => smithy_client_1.isa(o, "CopyObjectRequest");
})(CopyObjectRequest = exports.CopyObjectRequest || (exports.CopyObjectRequest = {}));
var CopyObjectResult;
(function (CopyObjectResult) {
    CopyObjectResult.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    CopyObjectResult.isa = (o) => smithy_client_1.isa(o, "CopyObjectResult");
})(CopyObjectResult = exports.CopyObjectResult || (exports.CopyObjectResult = {}));
var CopyPartResult;
(function (CopyPartResult) {
    CopyPartResult.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    CopyPartResult.isa = (o) => smithy_client_1.isa(o, "CopyPartResult");
})(CopyPartResult = exports.CopyPartResult || (exports.CopyPartResult = {}));
var CORSConfiguration;
(function (CORSConfiguration) {
    CORSConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    CORSConfiguration.isa = (o) => smithy_client_1.isa(o, "CORSConfiguration");
})(CORSConfiguration = exports.CORSConfiguration || (exports.CORSConfiguration = {}));
var CORSRule;
(function (CORSRule) {
    CORSRule.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    CORSRule.isa = (o) => smithy_client_1.isa(o, "CORSRule");
})(CORSRule = exports.CORSRule || (exports.CORSRule = {}));
var CreateBucketConfiguration;
(function (CreateBucketConfiguration) {
    CreateBucketConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    CreateBucketConfiguration.isa = (o) => smithy_client_1.isa(o, "CreateBucketConfiguration");
})(CreateBucketConfiguration = exports.CreateBucketConfiguration || (exports.CreateBucketConfiguration = {}));
var CreateBucketOutput;
(function (CreateBucketOutput) {
    CreateBucketOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    CreateBucketOutput.isa = (o) => smithy_client_1.isa(o, "CreateBucketOutput");
})(CreateBucketOutput = exports.CreateBucketOutput || (exports.CreateBucketOutput = {}));
var CreateBucketRequest;
(function (CreateBucketRequest) {
    CreateBucketRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    CreateBucketRequest.isa = (o) => smithy_client_1.isa(o, "CreateBucketRequest");
})(CreateBucketRequest = exports.CreateBucketRequest || (exports.CreateBucketRequest = {}));
var CreateMultipartUploadOutput;
(function (CreateMultipartUploadOutput) {
    CreateMultipartUploadOutput.filterSensitiveLog = (obj) => (Object.assign(Object.assign(Object.assign({}, obj), (obj.SSEKMSEncryptionContext && { SSEKMSEncryptionContext: smithy_client_1.SENSITIVE_STRING })), (obj.SSEKMSKeyId && { SSEKMSKeyId: smithy_client_1.SENSITIVE_STRING })));
    CreateMultipartUploadOutput.isa = (o) => smithy_client_1.isa(o, "CreateMultipartUploadOutput");
})(CreateMultipartUploadOutput = exports.CreateMultipartUploadOutput || (exports.CreateMultipartUploadOutput = {}));
var CreateMultipartUploadRequest;
(function (CreateMultipartUploadRequest) {
    CreateMultipartUploadRequest.filterSensitiveLog = (obj) => (Object.assign(Object.assign(Object.assign(Object.assign({}, obj), (obj.SSEKMSKeyId && { SSEKMSKeyId: smithy_client_1.SENSITIVE_STRING })), (obj.SSECustomerKey && { SSECustomerKey: smithy_client_1.SENSITIVE_STRING })), (obj.SSEKMSEncryptionContext && { SSEKMSEncryptionContext: smithy_client_1.SENSITIVE_STRING })));
    CreateMultipartUploadRequest.isa = (o) => smithy_client_1.isa(o, "CreateMultipartUploadRequest");
})(CreateMultipartUploadRequest = exports.CreateMultipartUploadRequest || (exports.CreateMultipartUploadRequest = {}));
var CSVInput;
(function (CSVInput) {
    CSVInput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    CSVInput.isa = (o) => smithy_client_1.isa(o, "CSVInput");
})(CSVInput = exports.CSVInput || (exports.CSVInput = {}));
var CSVOutput;
(function (CSVOutput) {
    CSVOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    CSVOutput.isa = (o) => smithy_client_1.isa(o, "CSVOutput");
})(CSVOutput = exports.CSVOutput || (exports.CSVOutput = {}));
var DefaultRetention;
(function (DefaultRetention) {
    DefaultRetention.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DefaultRetention.isa = (o) => smithy_client_1.isa(o, "DefaultRetention");
})(DefaultRetention = exports.DefaultRetention || (exports.DefaultRetention = {}));
var Delete;
(function (Delete) {
    Delete.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Delete.isa = (o) => smithy_client_1.isa(o, "Delete");
})(Delete = exports.Delete || (exports.Delete = {}));
var DeleteBucketAnalyticsConfigurationRequest;
(function (DeleteBucketAnalyticsConfigurationRequest) {
    DeleteBucketAnalyticsConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteBucketAnalyticsConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "DeleteBucketAnalyticsConfigurationRequest");
})(DeleteBucketAnalyticsConfigurationRequest = exports.DeleteBucketAnalyticsConfigurationRequest || (exports.DeleteBucketAnalyticsConfigurationRequest = {}));
var DeleteBucketCorsRequest;
(function (DeleteBucketCorsRequest) {
    DeleteBucketCorsRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteBucketCorsRequest.isa = (o) => smithy_client_1.isa(o, "DeleteBucketCorsRequest");
})(DeleteBucketCorsRequest = exports.DeleteBucketCorsRequest || (exports.DeleteBucketCorsRequest = {}));
var DeleteBucketEncryptionRequest;
(function (DeleteBucketEncryptionRequest) {
    DeleteBucketEncryptionRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteBucketEncryptionRequest.isa = (o) => smithy_client_1.isa(o, "DeleteBucketEncryptionRequest");
})(DeleteBucketEncryptionRequest = exports.DeleteBucketEncryptionRequest || (exports.DeleteBucketEncryptionRequest = {}));
var DeleteBucketInventoryConfigurationRequest;
(function (DeleteBucketInventoryConfigurationRequest) {
    DeleteBucketInventoryConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteBucketInventoryConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "DeleteBucketInventoryConfigurationRequest");
})(DeleteBucketInventoryConfigurationRequest = exports.DeleteBucketInventoryConfigurationRequest || (exports.DeleteBucketInventoryConfigurationRequest = {}));
var DeleteBucketLifecycleRequest;
(function (DeleteBucketLifecycleRequest) {
    DeleteBucketLifecycleRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteBucketLifecycleRequest.isa = (o) => smithy_client_1.isa(o, "DeleteBucketLifecycleRequest");
})(DeleteBucketLifecycleRequest = exports.DeleteBucketLifecycleRequest || (exports.DeleteBucketLifecycleRequest = {}));
var DeleteBucketMetricsConfigurationRequest;
(function (DeleteBucketMetricsConfigurationRequest) {
    DeleteBucketMetricsConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteBucketMetricsConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "DeleteBucketMetricsConfigurationRequest");
})(DeleteBucketMetricsConfigurationRequest = exports.DeleteBucketMetricsConfigurationRequest || (exports.DeleteBucketMetricsConfigurationRequest = {}));
var DeleteBucketPolicyRequest;
(function (DeleteBucketPolicyRequest) {
    DeleteBucketPolicyRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteBucketPolicyRequest.isa = (o) => smithy_client_1.isa(o, "DeleteBucketPolicyRequest");
})(DeleteBucketPolicyRequest = exports.DeleteBucketPolicyRequest || (exports.DeleteBucketPolicyRequest = {}));
var DeleteBucketReplicationRequest;
(function (DeleteBucketReplicationRequest) {
    DeleteBucketReplicationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteBucketReplicationRequest.isa = (o) => smithy_client_1.isa(o, "DeleteBucketReplicationRequest");
})(DeleteBucketReplicationRequest = exports.DeleteBucketReplicationRequest || (exports.DeleteBucketReplicationRequest = {}));
var DeleteBucketRequest;
(function (DeleteBucketRequest) {
    DeleteBucketRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteBucketRequest.isa = (o) => smithy_client_1.isa(o, "DeleteBucketRequest");
})(DeleteBucketRequest = exports.DeleteBucketRequest || (exports.DeleteBucketRequest = {}));
var DeleteBucketTaggingRequest;
(function (DeleteBucketTaggingRequest) {
    DeleteBucketTaggingRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteBucketTaggingRequest.isa = (o) => smithy_client_1.isa(o, "DeleteBucketTaggingRequest");
})(DeleteBucketTaggingRequest = exports.DeleteBucketTaggingRequest || (exports.DeleteBucketTaggingRequest = {}));
var DeleteBucketWebsiteRequest;
(function (DeleteBucketWebsiteRequest) {
    DeleteBucketWebsiteRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteBucketWebsiteRequest.isa = (o) => smithy_client_1.isa(o, "DeleteBucketWebsiteRequest");
})(DeleteBucketWebsiteRequest = exports.DeleteBucketWebsiteRequest || (exports.DeleteBucketWebsiteRequest = {}));
var DeletedObject;
(function (DeletedObject) {
    DeletedObject.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeletedObject.isa = (o) => smithy_client_1.isa(o, "DeletedObject");
})(DeletedObject = exports.DeletedObject || (exports.DeletedObject = {}));
var DeleteMarkerEntry;
(function (DeleteMarkerEntry) {
    DeleteMarkerEntry.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteMarkerEntry.isa = (o) => smithy_client_1.isa(o, "DeleteMarkerEntry");
})(DeleteMarkerEntry = exports.DeleteMarkerEntry || (exports.DeleteMarkerEntry = {}));
var DeleteMarkerReplication;
(function (DeleteMarkerReplication) {
    DeleteMarkerReplication.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteMarkerReplication.isa = (o) => smithy_client_1.isa(o, "DeleteMarkerReplication");
})(DeleteMarkerReplication = exports.DeleteMarkerReplication || (exports.DeleteMarkerReplication = {}));
var DeleteObjectOutput;
(function (DeleteObjectOutput) {
    DeleteObjectOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteObjectOutput.isa = (o) => smithy_client_1.isa(o, "DeleteObjectOutput");
})(DeleteObjectOutput = exports.DeleteObjectOutput || (exports.DeleteObjectOutput = {}));
var DeleteObjectRequest;
(function (DeleteObjectRequest) {
    DeleteObjectRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteObjectRequest.isa = (o) => smithy_client_1.isa(o, "DeleteObjectRequest");
})(DeleteObjectRequest = exports.DeleteObjectRequest || (exports.DeleteObjectRequest = {}));
var DeleteObjectsOutput;
(function (DeleteObjectsOutput) {
    DeleteObjectsOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteObjectsOutput.isa = (o) => smithy_client_1.isa(o, "DeleteObjectsOutput");
})(DeleteObjectsOutput = exports.DeleteObjectsOutput || (exports.DeleteObjectsOutput = {}));
var DeleteObjectsRequest;
(function (DeleteObjectsRequest) {
    DeleteObjectsRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteObjectsRequest.isa = (o) => smithy_client_1.isa(o, "DeleteObjectsRequest");
})(DeleteObjectsRequest = exports.DeleteObjectsRequest || (exports.DeleteObjectsRequest = {}));
var DeleteObjectTaggingOutput;
(function (DeleteObjectTaggingOutput) {
    DeleteObjectTaggingOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteObjectTaggingOutput.isa = (o) => smithy_client_1.isa(o, "DeleteObjectTaggingOutput");
})(DeleteObjectTaggingOutput = exports.DeleteObjectTaggingOutput || (exports.DeleteObjectTaggingOutput = {}));
var DeleteObjectTaggingRequest;
(function (DeleteObjectTaggingRequest) {
    DeleteObjectTaggingRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeleteObjectTaggingRequest.isa = (o) => smithy_client_1.isa(o, "DeleteObjectTaggingRequest");
})(DeleteObjectTaggingRequest = exports.DeleteObjectTaggingRequest || (exports.DeleteObjectTaggingRequest = {}));
var DeletePublicAccessBlockRequest;
(function (DeletePublicAccessBlockRequest) {
    DeletePublicAccessBlockRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    DeletePublicAccessBlockRequest.isa = (o) => smithy_client_1.isa(o, "DeletePublicAccessBlockRequest");
})(DeletePublicAccessBlockRequest = exports.DeletePublicAccessBlockRequest || (exports.DeletePublicAccessBlockRequest = {}));
var Destination;
(function (Destination) {
    Destination.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Destination.isa = (o) => smithy_client_1.isa(o, "Destination");
})(Destination = exports.Destination || (exports.Destination = {}));
var Encryption;
(function (Encryption) {
    Encryption.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.KMSKeyId && { KMSKeyId: smithy_client_1.SENSITIVE_STRING })));
    Encryption.isa = (o) => smithy_client_1.isa(o, "Encryption");
})(Encryption = exports.Encryption || (exports.Encryption = {}));
var EncryptionConfiguration;
(function (EncryptionConfiguration) {
    EncryptionConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    EncryptionConfiguration.isa = (o) => smithy_client_1.isa(o, "EncryptionConfiguration");
})(EncryptionConfiguration = exports.EncryptionConfiguration || (exports.EncryptionConfiguration = {}));
var EndEvent;
(function (EndEvent) {
    EndEvent.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    EndEvent.isa = (o) => smithy_client_1.isa(o, "EndEvent");
})(EndEvent = exports.EndEvent || (exports.EndEvent = {}));
var _Error;
(function (_Error) {
    _Error.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    _Error.isa = (o) => smithy_client_1.isa(o, "Error");
})(_Error = exports._Error || (exports._Error = {}));
var ErrorDocument;
(function (ErrorDocument) {
    ErrorDocument.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ErrorDocument.isa = (o) => smithy_client_1.isa(o, "ErrorDocument");
})(ErrorDocument = exports.ErrorDocument || (exports.ErrorDocument = {}));
var ExistingObjectReplication;
(function (ExistingObjectReplication) {
    ExistingObjectReplication.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ExistingObjectReplication.isa = (o) => smithy_client_1.isa(o, "ExistingObjectReplication");
})(ExistingObjectReplication = exports.ExistingObjectReplication || (exports.ExistingObjectReplication = {}));
var FileHeaderInfo;
(function (FileHeaderInfo) {
    FileHeaderInfo["IGNORE"] = "IGNORE";
    FileHeaderInfo["NONE"] = "NONE";
    FileHeaderInfo["USE"] = "USE";
})(FileHeaderInfo = exports.FileHeaderInfo || (exports.FileHeaderInfo = {}));
var FilterRule;
(function (FilterRule) {
    FilterRule.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    FilterRule.isa = (o) => smithy_client_1.isa(o, "FilterRule");
})(FilterRule = exports.FilterRule || (exports.FilterRule = {}));
var GetBucketAccelerateConfigurationOutput;
(function (GetBucketAccelerateConfigurationOutput) {
    GetBucketAccelerateConfigurationOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketAccelerateConfigurationOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketAccelerateConfigurationOutput");
})(GetBucketAccelerateConfigurationOutput = exports.GetBucketAccelerateConfigurationOutput || (exports.GetBucketAccelerateConfigurationOutput = {}));
var GetBucketAccelerateConfigurationRequest;
(function (GetBucketAccelerateConfigurationRequest) {
    GetBucketAccelerateConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketAccelerateConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketAccelerateConfigurationRequest");
})(GetBucketAccelerateConfigurationRequest = exports.GetBucketAccelerateConfigurationRequest || (exports.GetBucketAccelerateConfigurationRequest = {}));
var GetBucketAclOutput;
(function (GetBucketAclOutput) {
    GetBucketAclOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketAclOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketAclOutput");
})(GetBucketAclOutput = exports.GetBucketAclOutput || (exports.GetBucketAclOutput = {}));
var GetBucketAclRequest;
(function (GetBucketAclRequest) {
    GetBucketAclRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketAclRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketAclRequest");
})(GetBucketAclRequest = exports.GetBucketAclRequest || (exports.GetBucketAclRequest = {}));
var GetBucketAnalyticsConfigurationOutput;
(function (GetBucketAnalyticsConfigurationOutput) {
    GetBucketAnalyticsConfigurationOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketAnalyticsConfigurationOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketAnalyticsConfigurationOutput");
})(GetBucketAnalyticsConfigurationOutput = exports.GetBucketAnalyticsConfigurationOutput || (exports.GetBucketAnalyticsConfigurationOutput = {}));
var GetBucketAnalyticsConfigurationRequest;
(function (GetBucketAnalyticsConfigurationRequest) {
    GetBucketAnalyticsConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketAnalyticsConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketAnalyticsConfigurationRequest");
})(GetBucketAnalyticsConfigurationRequest = exports.GetBucketAnalyticsConfigurationRequest || (exports.GetBucketAnalyticsConfigurationRequest = {}));
var GetBucketCorsOutput;
(function (GetBucketCorsOutput) {
    GetBucketCorsOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketCorsOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketCorsOutput");
})(GetBucketCorsOutput = exports.GetBucketCorsOutput || (exports.GetBucketCorsOutput = {}));
var GetBucketCorsRequest;
(function (GetBucketCorsRequest) {
    GetBucketCorsRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketCorsRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketCorsRequest");
})(GetBucketCorsRequest = exports.GetBucketCorsRequest || (exports.GetBucketCorsRequest = {}));
var GetBucketEncryptionOutput;
(function (GetBucketEncryptionOutput) {
    GetBucketEncryptionOutput.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.ServerSideEncryptionConfiguration && {
        ServerSideEncryptionConfiguration: ServerSideEncryptionConfiguration.filterSensitiveLog(obj.ServerSideEncryptionConfiguration),
    })));
    GetBucketEncryptionOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketEncryptionOutput");
})(GetBucketEncryptionOutput = exports.GetBucketEncryptionOutput || (exports.GetBucketEncryptionOutput = {}));
var GetBucketEncryptionRequest;
(function (GetBucketEncryptionRequest) {
    GetBucketEncryptionRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketEncryptionRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketEncryptionRequest");
})(GetBucketEncryptionRequest = exports.GetBucketEncryptionRequest || (exports.GetBucketEncryptionRequest = {}));
var GetBucketInventoryConfigurationOutput;
(function (GetBucketInventoryConfigurationOutput) {
    GetBucketInventoryConfigurationOutput.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.InventoryConfiguration && {
        InventoryConfiguration: InventoryConfiguration.filterSensitiveLog(obj.InventoryConfiguration),
    })));
    GetBucketInventoryConfigurationOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketInventoryConfigurationOutput");
})(GetBucketInventoryConfigurationOutput = exports.GetBucketInventoryConfigurationOutput || (exports.GetBucketInventoryConfigurationOutput = {}));
var GetBucketInventoryConfigurationRequest;
(function (GetBucketInventoryConfigurationRequest) {
    GetBucketInventoryConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketInventoryConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketInventoryConfigurationRequest");
})(GetBucketInventoryConfigurationRequest = exports.GetBucketInventoryConfigurationRequest || (exports.GetBucketInventoryConfigurationRequest = {}));
var GetBucketLifecycleConfigurationOutput;
(function (GetBucketLifecycleConfigurationOutput) {
    GetBucketLifecycleConfigurationOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketLifecycleConfigurationOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketLifecycleConfigurationOutput");
})(GetBucketLifecycleConfigurationOutput = exports.GetBucketLifecycleConfigurationOutput || (exports.GetBucketLifecycleConfigurationOutput = {}));
var GetBucketLifecycleConfigurationRequest;
(function (GetBucketLifecycleConfigurationRequest) {
    GetBucketLifecycleConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketLifecycleConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketLifecycleConfigurationRequest");
})(GetBucketLifecycleConfigurationRequest = exports.GetBucketLifecycleConfigurationRequest || (exports.GetBucketLifecycleConfigurationRequest = {}));
var GetBucketLocationOutput;
(function (GetBucketLocationOutput) {
    GetBucketLocationOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketLocationOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketLocationOutput");
})(GetBucketLocationOutput = exports.GetBucketLocationOutput || (exports.GetBucketLocationOutput = {}));
var GetBucketLocationRequest;
(function (GetBucketLocationRequest) {
    GetBucketLocationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketLocationRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketLocationRequest");
})(GetBucketLocationRequest = exports.GetBucketLocationRequest || (exports.GetBucketLocationRequest = {}));
var GetBucketLoggingOutput;
(function (GetBucketLoggingOutput) {
    GetBucketLoggingOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketLoggingOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketLoggingOutput");
})(GetBucketLoggingOutput = exports.GetBucketLoggingOutput || (exports.GetBucketLoggingOutput = {}));
var GetBucketLoggingRequest;
(function (GetBucketLoggingRequest) {
    GetBucketLoggingRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketLoggingRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketLoggingRequest");
})(GetBucketLoggingRequest = exports.GetBucketLoggingRequest || (exports.GetBucketLoggingRequest = {}));
var GetBucketMetricsConfigurationOutput;
(function (GetBucketMetricsConfigurationOutput) {
    GetBucketMetricsConfigurationOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketMetricsConfigurationOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketMetricsConfigurationOutput");
})(GetBucketMetricsConfigurationOutput = exports.GetBucketMetricsConfigurationOutput || (exports.GetBucketMetricsConfigurationOutput = {}));
var GetBucketMetricsConfigurationRequest;
(function (GetBucketMetricsConfigurationRequest) {
    GetBucketMetricsConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketMetricsConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketMetricsConfigurationRequest");
})(GetBucketMetricsConfigurationRequest = exports.GetBucketMetricsConfigurationRequest || (exports.GetBucketMetricsConfigurationRequest = {}));
var GetBucketNotificationConfigurationRequest;
(function (GetBucketNotificationConfigurationRequest) {
    GetBucketNotificationConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketNotificationConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketNotificationConfigurationRequest");
})(GetBucketNotificationConfigurationRequest = exports.GetBucketNotificationConfigurationRequest || (exports.GetBucketNotificationConfigurationRequest = {}));
var GetBucketPolicyOutput;
(function (GetBucketPolicyOutput) {
    GetBucketPolicyOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketPolicyOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketPolicyOutput");
})(GetBucketPolicyOutput = exports.GetBucketPolicyOutput || (exports.GetBucketPolicyOutput = {}));
var GetBucketPolicyRequest;
(function (GetBucketPolicyRequest) {
    GetBucketPolicyRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketPolicyRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketPolicyRequest");
})(GetBucketPolicyRequest = exports.GetBucketPolicyRequest || (exports.GetBucketPolicyRequest = {}));
var GetBucketPolicyStatusOutput;
(function (GetBucketPolicyStatusOutput) {
    GetBucketPolicyStatusOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketPolicyStatusOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketPolicyStatusOutput");
})(GetBucketPolicyStatusOutput = exports.GetBucketPolicyStatusOutput || (exports.GetBucketPolicyStatusOutput = {}));
var GetBucketPolicyStatusRequest;
(function (GetBucketPolicyStatusRequest) {
    GetBucketPolicyStatusRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketPolicyStatusRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketPolicyStatusRequest");
})(GetBucketPolicyStatusRequest = exports.GetBucketPolicyStatusRequest || (exports.GetBucketPolicyStatusRequest = {}));
var GetBucketReplicationOutput;
(function (GetBucketReplicationOutput) {
    GetBucketReplicationOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketReplicationOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketReplicationOutput");
})(GetBucketReplicationOutput = exports.GetBucketReplicationOutput || (exports.GetBucketReplicationOutput = {}));
var GetBucketReplicationRequest;
(function (GetBucketReplicationRequest) {
    GetBucketReplicationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketReplicationRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketReplicationRequest");
})(GetBucketReplicationRequest = exports.GetBucketReplicationRequest || (exports.GetBucketReplicationRequest = {}));
var GetBucketRequestPaymentOutput;
(function (GetBucketRequestPaymentOutput) {
    GetBucketRequestPaymentOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketRequestPaymentOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketRequestPaymentOutput");
})(GetBucketRequestPaymentOutput = exports.GetBucketRequestPaymentOutput || (exports.GetBucketRequestPaymentOutput = {}));
var GetBucketRequestPaymentRequest;
(function (GetBucketRequestPaymentRequest) {
    GetBucketRequestPaymentRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketRequestPaymentRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketRequestPaymentRequest");
})(GetBucketRequestPaymentRequest = exports.GetBucketRequestPaymentRequest || (exports.GetBucketRequestPaymentRequest = {}));
var GetBucketTaggingOutput;
(function (GetBucketTaggingOutput) {
    GetBucketTaggingOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketTaggingOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketTaggingOutput");
})(GetBucketTaggingOutput = exports.GetBucketTaggingOutput || (exports.GetBucketTaggingOutput = {}));
var GetBucketTaggingRequest;
(function (GetBucketTaggingRequest) {
    GetBucketTaggingRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketTaggingRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketTaggingRequest");
})(GetBucketTaggingRequest = exports.GetBucketTaggingRequest || (exports.GetBucketTaggingRequest = {}));
var GetBucketVersioningOutput;
(function (GetBucketVersioningOutput) {
    GetBucketVersioningOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketVersioningOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketVersioningOutput");
})(GetBucketVersioningOutput = exports.GetBucketVersioningOutput || (exports.GetBucketVersioningOutput = {}));
var GetBucketVersioningRequest;
(function (GetBucketVersioningRequest) {
    GetBucketVersioningRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketVersioningRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketVersioningRequest");
})(GetBucketVersioningRequest = exports.GetBucketVersioningRequest || (exports.GetBucketVersioningRequest = {}));
var GetBucketWebsiteOutput;
(function (GetBucketWebsiteOutput) {
    GetBucketWebsiteOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketWebsiteOutput.isa = (o) => smithy_client_1.isa(o, "GetBucketWebsiteOutput");
})(GetBucketWebsiteOutput = exports.GetBucketWebsiteOutput || (exports.GetBucketWebsiteOutput = {}));
var GetBucketWebsiteRequest;
(function (GetBucketWebsiteRequest) {
    GetBucketWebsiteRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetBucketWebsiteRequest.isa = (o) => smithy_client_1.isa(o, "GetBucketWebsiteRequest");
})(GetBucketWebsiteRequest = exports.GetBucketWebsiteRequest || (exports.GetBucketWebsiteRequest = {}));
var GetObjectAclOutput;
(function (GetObjectAclOutput) {
    GetObjectAclOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetObjectAclOutput.isa = (o) => smithy_client_1.isa(o, "GetObjectAclOutput");
})(GetObjectAclOutput = exports.GetObjectAclOutput || (exports.GetObjectAclOutput = {}));
var GetObjectAclRequest;
(function (GetObjectAclRequest) {
    GetObjectAclRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetObjectAclRequest.isa = (o) => smithy_client_1.isa(o, "GetObjectAclRequest");
})(GetObjectAclRequest = exports.GetObjectAclRequest || (exports.GetObjectAclRequest = {}));
var GetObjectLegalHoldOutput;
(function (GetObjectLegalHoldOutput) {
    GetObjectLegalHoldOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetObjectLegalHoldOutput.isa = (o) => smithy_client_1.isa(o, "GetObjectLegalHoldOutput");
})(GetObjectLegalHoldOutput = exports.GetObjectLegalHoldOutput || (exports.GetObjectLegalHoldOutput = {}));
var GetObjectLegalHoldRequest;
(function (GetObjectLegalHoldRequest) {
    GetObjectLegalHoldRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetObjectLegalHoldRequest.isa = (o) => smithy_client_1.isa(o, "GetObjectLegalHoldRequest");
})(GetObjectLegalHoldRequest = exports.GetObjectLegalHoldRequest || (exports.GetObjectLegalHoldRequest = {}));
var GetObjectLockConfigurationOutput;
(function (GetObjectLockConfigurationOutput) {
    GetObjectLockConfigurationOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetObjectLockConfigurationOutput.isa = (o) => smithy_client_1.isa(o, "GetObjectLockConfigurationOutput");
})(GetObjectLockConfigurationOutput = exports.GetObjectLockConfigurationOutput || (exports.GetObjectLockConfigurationOutput = {}));
var GetObjectLockConfigurationRequest;
(function (GetObjectLockConfigurationRequest) {
    GetObjectLockConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetObjectLockConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "GetObjectLockConfigurationRequest");
})(GetObjectLockConfigurationRequest = exports.GetObjectLockConfigurationRequest || (exports.GetObjectLockConfigurationRequest = {}));
var GetObjectOutput;
(function (GetObjectOutput) {
    GetObjectOutput.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.SSEKMSKeyId && { SSEKMSKeyId: smithy_client_1.SENSITIVE_STRING })));
    GetObjectOutput.isa = (o) => smithy_client_1.isa(o, "GetObjectOutput");
})(GetObjectOutput = exports.GetObjectOutput || (exports.GetObjectOutput = {}));
var GetObjectRequest;
(function (GetObjectRequest) {
    GetObjectRequest.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.SSECustomerKey && { SSECustomerKey: smithy_client_1.SENSITIVE_STRING })));
    GetObjectRequest.isa = (o) => smithy_client_1.isa(o, "GetObjectRequest");
})(GetObjectRequest = exports.GetObjectRequest || (exports.GetObjectRequest = {}));
var GetObjectRetentionOutput;
(function (GetObjectRetentionOutput) {
    GetObjectRetentionOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetObjectRetentionOutput.isa = (o) => smithy_client_1.isa(o, "GetObjectRetentionOutput");
})(GetObjectRetentionOutput = exports.GetObjectRetentionOutput || (exports.GetObjectRetentionOutput = {}));
var GetObjectRetentionRequest;
(function (GetObjectRetentionRequest) {
    GetObjectRetentionRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetObjectRetentionRequest.isa = (o) => smithy_client_1.isa(o, "GetObjectRetentionRequest");
})(GetObjectRetentionRequest = exports.GetObjectRetentionRequest || (exports.GetObjectRetentionRequest = {}));
var GetObjectTaggingOutput;
(function (GetObjectTaggingOutput) {
    GetObjectTaggingOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetObjectTaggingOutput.isa = (o) => smithy_client_1.isa(o, "GetObjectTaggingOutput");
})(GetObjectTaggingOutput = exports.GetObjectTaggingOutput || (exports.GetObjectTaggingOutput = {}));
var GetObjectTaggingRequest;
(function (GetObjectTaggingRequest) {
    GetObjectTaggingRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetObjectTaggingRequest.isa = (o) => smithy_client_1.isa(o, "GetObjectTaggingRequest");
})(GetObjectTaggingRequest = exports.GetObjectTaggingRequest || (exports.GetObjectTaggingRequest = {}));
var GetObjectTorrentOutput;
(function (GetObjectTorrentOutput) {
    GetObjectTorrentOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetObjectTorrentOutput.isa = (o) => smithy_client_1.isa(o, "GetObjectTorrentOutput");
})(GetObjectTorrentOutput = exports.GetObjectTorrentOutput || (exports.GetObjectTorrentOutput = {}));
var GetObjectTorrentRequest;
(function (GetObjectTorrentRequest) {
    GetObjectTorrentRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetObjectTorrentRequest.isa = (o) => smithy_client_1.isa(o, "GetObjectTorrentRequest");
})(GetObjectTorrentRequest = exports.GetObjectTorrentRequest || (exports.GetObjectTorrentRequest = {}));
var GetPublicAccessBlockOutput;
(function (GetPublicAccessBlockOutput) {
    GetPublicAccessBlockOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetPublicAccessBlockOutput.isa = (o) => smithy_client_1.isa(o, "GetPublicAccessBlockOutput");
})(GetPublicAccessBlockOutput = exports.GetPublicAccessBlockOutput || (exports.GetPublicAccessBlockOutput = {}));
var GetPublicAccessBlockRequest;
(function (GetPublicAccessBlockRequest) {
    GetPublicAccessBlockRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GetPublicAccessBlockRequest.isa = (o) => smithy_client_1.isa(o, "GetPublicAccessBlockRequest");
})(GetPublicAccessBlockRequest = exports.GetPublicAccessBlockRequest || (exports.GetPublicAccessBlockRequest = {}));
var GlacierJobParameters;
(function (GlacierJobParameters) {
    GlacierJobParameters.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    GlacierJobParameters.isa = (o) => smithy_client_1.isa(o, "GlacierJobParameters");
})(GlacierJobParameters = exports.GlacierJobParameters || (exports.GlacierJobParameters = {}));
var Grant;
(function (Grant) {
    Grant.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Grant.isa = (o) => smithy_client_1.isa(o, "Grant");
})(Grant = exports.Grant || (exports.Grant = {}));
var Grantee;
(function (Grantee) {
    Grantee.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Grantee.isa = (o) => smithy_client_1.isa(o, "Grantee");
})(Grantee = exports.Grantee || (exports.Grantee = {}));
var HeadBucketRequest;
(function (HeadBucketRequest) {
    HeadBucketRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    HeadBucketRequest.isa = (o) => smithy_client_1.isa(o, "HeadBucketRequest");
})(HeadBucketRequest = exports.HeadBucketRequest || (exports.HeadBucketRequest = {}));
var HeadObjectOutput;
(function (HeadObjectOutput) {
    HeadObjectOutput.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.SSEKMSKeyId && { SSEKMSKeyId: smithy_client_1.SENSITIVE_STRING })));
    HeadObjectOutput.isa = (o) => smithy_client_1.isa(o, "HeadObjectOutput");
})(HeadObjectOutput = exports.HeadObjectOutput || (exports.HeadObjectOutput = {}));
var HeadObjectRequest;
(function (HeadObjectRequest) {
    HeadObjectRequest.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.SSECustomerKey && { SSECustomerKey: smithy_client_1.SENSITIVE_STRING })));
    HeadObjectRequest.isa = (o) => smithy_client_1.isa(o, "HeadObjectRequest");
})(HeadObjectRequest = exports.HeadObjectRequest || (exports.HeadObjectRequest = {}));
var IndexDocument;
(function (IndexDocument) {
    IndexDocument.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    IndexDocument.isa = (o) => smithy_client_1.isa(o, "IndexDocument");
})(IndexDocument = exports.IndexDocument || (exports.IndexDocument = {}));
var Initiator;
(function (Initiator) {
    Initiator.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Initiator.isa = (o) => smithy_client_1.isa(o, "Initiator");
})(Initiator = exports.Initiator || (exports.Initiator = {}));
var InputSerialization;
(function (InputSerialization) {
    InputSerialization.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    InputSerialization.isa = (o) => smithy_client_1.isa(o, "InputSerialization");
})(InputSerialization = exports.InputSerialization || (exports.InputSerialization = {}));
var InventoryConfiguration;
(function (InventoryConfiguration) {
    InventoryConfiguration.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.Destination && { Destination: InventoryDestination.filterSensitiveLog(obj.Destination) })));
    InventoryConfiguration.isa = (o) => smithy_client_1.isa(o, "InventoryConfiguration");
})(InventoryConfiguration = exports.InventoryConfiguration || (exports.InventoryConfiguration = {}));
var InventoryDestination;
(function (InventoryDestination) {
    InventoryDestination.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.S3BucketDestination && {
        S3BucketDestination: InventoryS3BucketDestination.filterSensitiveLog(obj.S3BucketDestination),
    })));
    InventoryDestination.isa = (o) => smithy_client_1.isa(o, "InventoryDestination");
})(InventoryDestination = exports.InventoryDestination || (exports.InventoryDestination = {}));
var InventoryEncryption;
(function (InventoryEncryption) {
    InventoryEncryption.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.SSEKMS && { SSEKMS: SSEKMS.filterSensitiveLog(obj.SSEKMS) })));
    InventoryEncryption.isa = (o) => smithy_client_1.isa(o, "InventoryEncryption");
})(InventoryEncryption = exports.InventoryEncryption || (exports.InventoryEncryption = {}));
var InventoryFilter;
(function (InventoryFilter) {
    InventoryFilter.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    InventoryFilter.isa = (o) => smithy_client_1.isa(o, "InventoryFilter");
})(InventoryFilter = exports.InventoryFilter || (exports.InventoryFilter = {}));
var InventoryS3BucketDestination;
(function (InventoryS3BucketDestination) {
    InventoryS3BucketDestination.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.Encryption && { Encryption: InventoryEncryption.filterSensitiveLog(obj.Encryption) })));
    InventoryS3BucketDestination.isa = (o) => smithy_client_1.isa(o, "InventoryS3BucketDestination");
})(InventoryS3BucketDestination = exports.InventoryS3BucketDestination || (exports.InventoryS3BucketDestination = {}));
var InventorySchedule;
(function (InventorySchedule) {
    InventorySchedule.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    InventorySchedule.isa = (o) => smithy_client_1.isa(o, "InventorySchedule");
})(InventorySchedule = exports.InventorySchedule || (exports.InventorySchedule = {}));
var JSONInput;
(function (JSONInput) {
    JSONInput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    JSONInput.isa = (o) => smithy_client_1.isa(o, "JSONInput");
})(JSONInput = exports.JSONInput || (exports.JSONInput = {}));
var JSONOutput;
(function (JSONOutput) {
    JSONOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    JSONOutput.isa = (o) => smithy_client_1.isa(o, "JSONOutput");
})(JSONOutput = exports.JSONOutput || (exports.JSONOutput = {}));
var JSONType;
(function (JSONType) {
    JSONType["DOCUMENT"] = "DOCUMENT";
    JSONType["LINES"] = "LINES";
})(JSONType = exports.JSONType || (exports.JSONType = {}));
var LambdaFunctionConfiguration;
(function (LambdaFunctionConfiguration) {
    LambdaFunctionConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    LambdaFunctionConfiguration.isa = (o) => smithy_client_1.isa(o, "LambdaFunctionConfiguration");
})(LambdaFunctionConfiguration = exports.LambdaFunctionConfiguration || (exports.LambdaFunctionConfiguration = {}));
var LifecycleExpiration;
(function (LifecycleExpiration) {
    LifecycleExpiration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    LifecycleExpiration.isa = (o) => smithy_client_1.isa(o, "LifecycleExpiration");
})(LifecycleExpiration = exports.LifecycleExpiration || (exports.LifecycleExpiration = {}));
var LifecycleRule;
(function (LifecycleRule) {
    LifecycleRule.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    LifecycleRule.isa = (o) => smithy_client_1.isa(o, "LifecycleRule");
})(LifecycleRule = exports.LifecycleRule || (exports.LifecycleRule = {}));
var LifecycleRuleAndOperator;
(function (LifecycleRuleAndOperator) {
    LifecycleRuleAndOperator.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    LifecycleRuleAndOperator.isa = (o) => smithy_client_1.isa(o, "LifecycleRuleAndOperator");
})(LifecycleRuleAndOperator = exports.LifecycleRuleAndOperator || (exports.LifecycleRuleAndOperator = {}));
var LifecycleRuleFilter;
(function (LifecycleRuleFilter) {
    LifecycleRuleFilter.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    LifecycleRuleFilter.isa = (o) => smithy_client_1.isa(o, "LifecycleRuleFilter");
})(LifecycleRuleFilter = exports.LifecycleRuleFilter || (exports.LifecycleRuleFilter = {}));
var ListBucketAnalyticsConfigurationsOutput;
(function (ListBucketAnalyticsConfigurationsOutput) {
    ListBucketAnalyticsConfigurationsOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListBucketAnalyticsConfigurationsOutput.isa = (o) => smithy_client_1.isa(o, "ListBucketAnalyticsConfigurationsOutput");
})(ListBucketAnalyticsConfigurationsOutput = exports.ListBucketAnalyticsConfigurationsOutput || (exports.ListBucketAnalyticsConfigurationsOutput = {}));
var ListBucketAnalyticsConfigurationsRequest;
(function (ListBucketAnalyticsConfigurationsRequest) {
    ListBucketAnalyticsConfigurationsRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListBucketAnalyticsConfigurationsRequest.isa = (o) => smithy_client_1.isa(o, "ListBucketAnalyticsConfigurationsRequest");
})(ListBucketAnalyticsConfigurationsRequest = exports.ListBucketAnalyticsConfigurationsRequest || (exports.ListBucketAnalyticsConfigurationsRequest = {}));
var ListBucketInventoryConfigurationsOutput;
(function (ListBucketInventoryConfigurationsOutput) {
    ListBucketInventoryConfigurationsOutput.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.InventoryConfigurationList && {
        InventoryConfigurationList: obj.InventoryConfigurationList.map((item) => InventoryConfiguration.filterSensitiveLog(item)),
    })));
    ListBucketInventoryConfigurationsOutput.isa = (o) => smithy_client_1.isa(o, "ListBucketInventoryConfigurationsOutput");
})(ListBucketInventoryConfigurationsOutput = exports.ListBucketInventoryConfigurationsOutput || (exports.ListBucketInventoryConfigurationsOutput = {}));
var ListBucketInventoryConfigurationsRequest;
(function (ListBucketInventoryConfigurationsRequest) {
    ListBucketInventoryConfigurationsRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListBucketInventoryConfigurationsRequest.isa = (o) => smithy_client_1.isa(o, "ListBucketInventoryConfigurationsRequest");
})(ListBucketInventoryConfigurationsRequest = exports.ListBucketInventoryConfigurationsRequest || (exports.ListBucketInventoryConfigurationsRequest = {}));
var ListBucketMetricsConfigurationsOutput;
(function (ListBucketMetricsConfigurationsOutput) {
    ListBucketMetricsConfigurationsOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListBucketMetricsConfigurationsOutput.isa = (o) => smithy_client_1.isa(o, "ListBucketMetricsConfigurationsOutput");
})(ListBucketMetricsConfigurationsOutput = exports.ListBucketMetricsConfigurationsOutput || (exports.ListBucketMetricsConfigurationsOutput = {}));
var ListBucketMetricsConfigurationsRequest;
(function (ListBucketMetricsConfigurationsRequest) {
    ListBucketMetricsConfigurationsRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListBucketMetricsConfigurationsRequest.isa = (o) => smithy_client_1.isa(o, "ListBucketMetricsConfigurationsRequest");
})(ListBucketMetricsConfigurationsRequest = exports.ListBucketMetricsConfigurationsRequest || (exports.ListBucketMetricsConfigurationsRequest = {}));
var ListBucketsOutput;
(function (ListBucketsOutput) {
    ListBucketsOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListBucketsOutput.isa = (o) => smithy_client_1.isa(o, "ListBucketsOutput");
})(ListBucketsOutput = exports.ListBucketsOutput || (exports.ListBucketsOutput = {}));
var ListMultipartUploadsOutput;
(function (ListMultipartUploadsOutput) {
    ListMultipartUploadsOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListMultipartUploadsOutput.isa = (o) => smithy_client_1.isa(o, "ListMultipartUploadsOutput");
})(ListMultipartUploadsOutput = exports.ListMultipartUploadsOutput || (exports.ListMultipartUploadsOutput = {}));
var ListMultipartUploadsRequest;
(function (ListMultipartUploadsRequest) {
    ListMultipartUploadsRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListMultipartUploadsRequest.isa = (o) => smithy_client_1.isa(o, "ListMultipartUploadsRequest");
})(ListMultipartUploadsRequest = exports.ListMultipartUploadsRequest || (exports.ListMultipartUploadsRequest = {}));
var ListObjectsOutput;
(function (ListObjectsOutput) {
    ListObjectsOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListObjectsOutput.isa = (o) => smithy_client_1.isa(o, "ListObjectsOutput");
})(ListObjectsOutput = exports.ListObjectsOutput || (exports.ListObjectsOutput = {}));
var ListObjectsRequest;
(function (ListObjectsRequest) {
    ListObjectsRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListObjectsRequest.isa = (o) => smithy_client_1.isa(o, "ListObjectsRequest");
})(ListObjectsRequest = exports.ListObjectsRequest || (exports.ListObjectsRequest = {}));
var ListObjectsV2Output;
(function (ListObjectsV2Output) {
    ListObjectsV2Output.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListObjectsV2Output.isa = (o) => smithy_client_1.isa(o, "ListObjectsV2Output");
})(ListObjectsV2Output = exports.ListObjectsV2Output || (exports.ListObjectsV2Output = {}));
var ListObjectsV2Request;
(function (ListObjectsV2Request) {
    ListObjectsV2Request.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListObjectsV2Request.isa = (o) => smithy_client_1.isa(o, "ListObjectsV2Request");
})(ListObjectsV2Request = exports.ListObjectsV2Request || (exports.ListObjectsV2Request = {}));
var ListObjectVersionsOutput;
(function (ListObjectVersionsOutput) {
    ListObjectVersionsOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListObjectVersionsOutput.isa = (o) => smithy_client_1.isa(o, "ListObjectVersionsOutput");
})(ListObjectVersionsOutput = exports.ListObjectVersionsOutput || (exports.ListObjectVersionsOutput = {}));
var ListObjectVersionsRequest;
(function (ListObjectVersionsRequest) {
    ListObjectVersionsRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListObjectVersionsRequest.isa = (o) => smithy_client_1.isa(o, "ListObjectVersionsRequest");
})(ListObjectVersionsRequest = exports.ListObjectVersionsRequest || (exports.ListObjectVersionsRequest = {}));
var ListPartsOutput;
(function (ListPartsOutput) {
    ListPartsOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListPartsOutput.isa = (o) => smithy_client_1.isa(o, "ListPartsOutput");
})(ListPartsOutput = exports.ListPartsOutput || (exports.ListPartsOutput = {}));
var ListPartsRequest;
(function (ListPartsRequest) {
    ListPartsRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ListPartsRequest.isa = (o) => smithy_client_1.isa(o, "ListPartsRequest");
})(ListPartsRequest = exports.ListPartsRequest || (exports.ListPartsRequest = {}));
var LoggingEnabled;
(function (LoggingEnabled) {
    LoggingEnabled.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    LoggingEnabled.isa = (o) => smithy_client_1.isa(o, "LoggingEnabled");
})(LoggingEnabled = exports.LoggingEnabled || (exports.LoggingEnabled = {}));
var MetadataEntry;
(function (MetadataEntry) {
    MetadataEntry.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    MetadataEntry.isa = (o) => smithy_client_1.isa(o, "MetadataEntry");
})(MetadataEntry = exports.MetadataEntry || (exports.MetadataEntry = {}));
var Metrics;
(function (Metrics) {
    Metrics.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Metrics.isa = (o) => smithy_client_1.isa(o, "Metrics");
})(Metrics = exports.Metrics || (exports.Metrics = {}));
var MetricsAndOperator;
(function (MetricsAndOperator) {
    MetricsAndOperator.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    MetricsAndOperator.isa = (o) => smithy_client_1.isa(o, "MetricsAndOperator");
})(MetricsAndOperator = exports.MetricsAndOperator || (exports.MetricsAndOperator = {}));
var MetricsConfiguration;
(function (MetricsConfiguration) {
    MetricsConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    MetricsConfiguration.isa = (o) => smithy_client_1.isa(o, "MetricsConfiguration");
})(MetricsConfiguration = exports.MetricsConfiguration || (exports.MetricsConfiguration = {}));
var MetricsFilter;
(function (MetricsFilter) {
    MetricsFilter.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    MetricsFilter.isa = (o) => smithy_client_1.isa(o, "MetricsFilter");
})(MetricsFilter = exports.MetricsFilter || (exports.MetricsFilter = {}));
var MultipartUpload;
(function (MultipartUpload) {
    MultipartUpload.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    MultipartUpload.isa = (o) => smithy_client_1.isa(o, "MultipartUpload");
})(MultipartUpload = exports.MultipartUpload || (exports.MultipartUpload = {}));
var NoncurrentVersionExpiration;
(function (NoncurrentVersionExpiration) {
    NoncurrentVersionExpiration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    NoncurrentVersionExpiration.isa = (o) => smithy_client_1.isa(o, "NoncurrentVersionExpiration");
})(NoncurrentVersionExpiration = exports.NoncurrentVersionExpiration || (exports.NoncurrentVersionExpiration = {}));
var NoncurrentVersionTransition;
(function (NoncurrentVersionTransition) {
    NoncurrentVersionTransition.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    NoncurrentVersionTransition.isa = (o) => smithy_client_1.isa(o, "NoncurrentVersionTransition");
})(NoncurrentVersionTransition = exports.NoncurrentVersionTransition || (exports.NoncurrentVersionTransition = {}));
var NoSuchBucket;
(function (NoSuchBucket) {
    NoSuchBucket.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    NoSuchBucket.isa = (o) => smithy_client_1.isa(o, "NoSuchBucket");
})(NoSuchBucket = exports.NoSuchBucket || (exports.NoSuchBucket = {}));
var NoSuchKey;
(function (NoSuchKey) {
    NoSuchKey.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    NoSuchKey.isa = (o) => smithy_client_1.isa(o, "NoSuchKey");
})(NoSuchKey = exports.NoSuchKey || (exports.NoSuchKey = {}));
var NoSuchUpload;
(function (NoSuchUpload) {
    NoSuchUpload.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    NoSuchUpload.isa = (o) => smithy_client_1.isa(o, "NoSuchUpload");
})(NoSuchUpload = exports.NoSuchUpload || (exports.NoSuchUpload = {}));
var NotificationConfiguration;
(function (NotificationConfiguration) {
    NotificationConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    NotificationConfiguration.isa = (o) => smithy_client_1.isa(o, "NotificationConfiguration");
})(NotificationConfiguration = exports.NotificationConfiguration || (exports.NotificationConfiguration = {}));
var NotificationConfigurationFilter;
(function (NotificationConfigurationFilter) {
    NotificationConfigurationFilter.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    NotificationConfigurationFilter.isa = (o) => smithy_client_1.isa(o, "NotificationConfigurationFilter");
})(NotificationConfigurationFilter = exports.NotificationConfigurationFilter || (exports.NotificationConfigurationFilter = {}));
var _Object;
(function (_Object) {
    _Object.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    _Object.isa = (o) => smithy_client_1.isa(o, "Object");
})(_Object = exports._Object || (exports._Object = {}));
var ObjectAlreadyInActiveTierError;
(function (ObjectAlreadyInActiveTierError) {
    ObjectAlreadyInActiveTierError.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ObjectAlreadyInActiveTierError.isa = (o) => smithy_client_1.isa(o, "ObjectAlreadyInActiveTierError");
})(ObjectAlreadyInActiveTierError = exports.ObjectAlreadyInActiveTierError || (exports.ObjectAlreadyInActiveTierError = {}));
var ObjectIdentifier;
(function (ObjectIdentifier) {
    ObjectIdentifier.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ObjectIdentifier.isa = (o) => smithy_client_1.isa(o, "ObjectIdentifier");
})(ObjectIdentifier = exports.ObjectIdentifier || (exports.ObjectIdentifier = {}));
var ObjectLockConfiguration;
(function (ObjectLockConfiguration) {
    ObjectLockConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ObjectLockConfiguration.isa = (o) => smithy_client_1.isa(o, "ObjectLockConfiguration");
})(ObjectLockConfiguration = exports.ObjectLockConfiguration || (exports.ObjectLockConfiguration = {}));
var ObjectLockLegalHold;
(function (ObjectLockLegalHold) {
    ObjectLockLegalHold.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ObjectLockLegalHold.isa = (o) => smithy_client_1.isa(o, "ObjectLockLegalHold");
})(ObjectLockLegalHold = exports.ObjectLockLegalHold || (exports.ObjectLockLegalHold = {}));
var ObjectLockRetention;
(function (ObjectLockRetention) {
    ObjectLockRetention.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ObjectLockRetention.isa = (o) => smithy_client_1.isa(o, "ObjectLockRetention");
})(ObjectLockRetention = exports.ObjectLockRetention || (exports.ObjectLockRetention = {}));
var ObjectLockRule;
(function (ObjectLockRule) {
    ObjectLockRule.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ObjectLockRule.isa = (o) => smithy_client_1.isa(o, "ObjectLockRule");
})(ObjectLockRule = exports.ObjectLockRule || (exports.ObjectLockRule = {}));
var ObjectNotInActiveTierError;
(function (ObjectNotInActiveTierError) {
    ObjectNotInActiveTierError.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ObjectNotInActiveTierError.isa = (o) => smithy_client_1.isa(o, "ObjectNotInActiveTierError");
})(ObjectNotInActiveTierError = exports.ObjectNotInActiveTierError || (exports.ObjectNotInActiveTierError = {}));
var ObjectVersion;
(function (ObjectVersion) {
    ObjectVersion.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ObjectVersion.isa = (o) => smithy_client_1.isa(o, "ObjectVersion");
})(ObjectVersion = exports.ObjectVersion || (exports.ObjectVersion = {}));
var OutputLocation;
(function (OutputLocation) {
    OutputLocation.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.S3 && { S3: S3Location.filterSensitiveLog(obj.S3) })));
    OutputLocation.isa = (o) => smithy_client_1.isa(o, "OutputLocation");
})(OutputLocation = exports.OutputLocation || (exports.OutputLocation = {}));
var OutputSerialization;
(function (OutputSerialization) {
    OutputSerialization.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    OutputSerialization.isa = (o) => smithy_client_1.isa(o, "OutputSerialization");
})(OutputSerialization = exports.OutputSerialization || (exports.OutputSerialization = {}));
var Owner;
(function (Owner) {
    Owner.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Owner.isa = (o) => smithy_client_1.isa(o, "Owner");
})(Owner = exports.Owner || (exports.Owner = {}));
var ParquetInput;
(function (ParquetInput) {
    ParquetInput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ParquetInput.isa = (o) => smithy_client_1.isa(o, "ParquetInput");
})(ParquetInput = exports.ParquetInput || (exports.ParquetInput = {}));
var Part;
(function (Part) {
    Part.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Part.isa = (o) => smithy_client_1.isa(o, "Part");
})(Part = exports.Part || (exports.Part = {}));
var PolicyStatus;
(function (PolicyStatus) {
    PolicyStatus.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PolicyStatus.isa = (o) => smithy_client_1.isa(o, "PolicyStatus");
})(PolicyStatus = exports.PolicyStatus || (exports.PolicyStatus = {}));
var Progress;
(function (Progress) {
    Progress.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Progress.isa = (o) => smithy_client_1.isa(o, "Progress");
})(Progress = exports.Progress || (exports.Progress = {}));
var ProgressEvent;
(function (ProgressEvent) {
    ProgressEvent.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ProgressEvent.isa = (o) => smithy_client_1.isa(o, "ProgressEvent");
})(ProgressEvent = exports.ProgressEvent || (exports.ProgressEvent = {}));
var PublicAccessBlockConfiguration;
(function (PublicAccessBlockConfiguration) {
    PublicAccessBlockConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PublicAccessBlockConfiguration.isa = (o) => smithy_client_1.isa(o, "PublicAccessBlockConfiguration");
})(PublicAccessBlockConfiguration = exports.PublicAccessBlockConfiguration || (exports.PublicAccessBlockConfiguration = {}));
var PutBucketAccelerateConfigurationRequest;
(function (PutBucketAccelerateConfigurationRequest) {
    PutBucketAccelerateConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketAccelerateConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketAccelerateConfigurationRequest");
})(PutBucketAccelerateConfigurationRequest = exports.PutBucketAccelerateConfigurationRequest || (exports.PutBucketAccelerateConfigurationRequest = {}));
var PutBucketAclRequest;
(function (PutBucketAclRequest) {
    PutBucketAclRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketAclRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketAclRequest");
})(PutBucketAclRequest = exports.PutBucketAclRequest || (exports.PutBucketAclRequest = {}));
var PutBucketAnalyticsConfigurationRequest;
(function (PutBucketAnalyticsConfigurationRequest) {
    PutBucketAnalyticsConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketAnalyticsConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketAnalyticsConfigurationRequest");
})(PutBucketAnalyticsConfigurationRequest = exports.PutBucketAnalyticsConfigurationRequest || (exports.PutBucketAnalyticsConfigurationRequest = {}));
var PutBucketCorsRequest;
(function (PutBucketCorsRequest) {
    PutBucketCorsRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketCorsRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketCorsRequest");
})(PutBucketCorsRequest = exports.PutBucketCorsRequest || (exports.PutBucketCorsRequest = {}));
var PutBucketEncryptionRequest;
(function (PutBucketEncryptionRequest) {
    PutBucketEncryptionRequest.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.ServerSideEncryptionConfiguration && {
        ServerSideEncryptionConfiguration: ServerSideEncryptionConfiguration.filterSensitiveLog(obj.ServerSideEncryptionConfiguration),
    })));
    PutBucketEncryptionRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketEncryptionRequest");
})(PutBucketEncryptionRequest = exports.PutBucketEncryptionRequest || (exports.PutBucketEncryptionRequest = {}));
var PutBucketInventoryConfigurationRequest;
(function (PutBucketInventoryConfigurationRequest) {
    PutBucketInventoryConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.InventoryConfiguration && {
        InventoryConfiguration: InventoryConfiguration.filterSensitiveLog(obj.InventoryConfiguration),
    })));
    PutBucketInventoryConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketInventoryConfigurationRequest");
})(PutBucketInventoryConfigurationRequest = exports.PutBucketInventoryConfigurationRequest || (exports.PutBucketInventoryConfigurationRequest = {}));
var PutBucketLifecycleConfigurationRequest;
(function (PutBucketLifecycleConfigurationRequest) {
    PutBucketLifecycleConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketLifecycleConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketLifecycleConfigurationRequest");
})(PutBucketLifecycleConfigurationRequest = exports.PutBucketLifecycleConfigurationRequest || (exports.PutBucketLifecycleConfigurationRequest = {}));
var PutBucketLoggingRequest;
(function (PutBucketLoggingRequest) {
    PutBucketLoggingRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketLoggingRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketLoggingRequest");
})(PutBucketLoggingRequest = exports.PutBucketLoggingRequest || (exports.PutBucketLoggingRequest = {}));
var PutBucketMetricsConfigurationRequest;
(function (PutBucketMetricsConfigurationRequest) {
    PutBucketMetricsConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketMetricsConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketMetricsConfigurationRequest");
})(PutBucketMetricsConfigurationRequest = exports.PutBucketMetricsConfigurationRequest || (exports.PutBucketMetricsConfigurationRequest = {}));
var PutBucketNotificationConfigurationRequest;
(function (PutBucketNotificationConfigurationRequest) {
    PutBucketNotificationConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketNotificationConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketNotificationConfigurationRequest");
})(PutBucketNotificationConfigurationRequest = exports.PutBucketNotificationConfigurationRequest || (exports.PutBucketNotificationConfigurationRequest = {}));
var PutBucketPolicyRequest;
(function (PutBucketPolicyRequest) {
    PutBucketPolicyRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketPolicyRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketPolicyRequest");
})(PutBucketPolicyRequest = exports.PutBucketPolicyRequest || (exports.PutBucketPolicyRequest = {}));
var PutBucketReplicationRequest;
(function (PutBucketReplicationRequest) {
    PutBucketReplicationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketReplicationRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketReplicationRequest");
})(PutBucketReplicationRequest = exports.PutBucketReplicationRequest || (exports.PutBucketReplicationRequest = {}));
var PutBucketRequestPaymentRequest;
(function (PutBucketRequestPaymentRequest) {
    PutBucketRequestPaymentRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketRequestPaymentRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketRequestPaymentRequest");
})(PutBucketRequestPaymentRequest = exports.PutBucketRequestPaymentRequest || (exports.PutBucketRequestPaymentRequest = {}));
var PutBucketTaggingRequest;
(function (PutBucketTaggingRequest) {
    PutBucketTaggingRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketTaggingRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketTaggingRequest");
})(PutBucketTaggingRequest = exports.PutBucketTaggingRequest || (exports.PutBucketTaggingRequest = {}));
var PutBucketVersioningRequest;
(function (PutBucketVersioningRequest) {
    PutBucketVersioningRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketVersioningRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketVersioningRequest");
})(PutBucketVersioningRequest = exports.PutBucketVersioningRequest || (exports.PutBucketVersioningRequest = {}));
var PutBucketWebsiteRequest;
(function (PutBucketWebsiteRequest) {
    PutBucketWebsiteRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutBucketWebsiteRequest.isa = (o) => smithy_client_1.isa(o, "PutBucketWebsiteRequest");
})(PutBucketWebsiteRequest = exports.PutBucketWebsiteRequest || (exports.PutBucketWebsiteRequest = {}));
var PutObjectAclOutput;
(function (PutObjectAclOutput) {
    PutObjectAclOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutObjectAclOutput.isa = (o) => smithy_client_1.isa(o, "PutObjectAclOutput");
})(PutObjectAclOutput = exports.PutObjectAclOutput || (exports.PutObjectAclOutput = {}));
var PutObjectAclRequest;
(function (PutObjectAclRequest) {
    PutObjectAclRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutObjectAclRequest.isa = (o) => smithy_client_1.isa(o, "PutObjectAclRequest");
})(PutObjectAclRequest = exports.PutObjectAclRequest || (exports.PutObjectAclRequest = {}));
var PutObjectLegalHoldOutput;
(function (PutObjectLegalHoldOutput) {
    PutObjectLegalHoldOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutObjectLegalHoldOutput.isa = (o) => smithy_client_1.isa(o, "PutObjectLegalHoldOutput");
})(PutObjectLegalHoldOutput = exports.PutObjectLegalHoldOutput || (exports.PutObjectLegalHoldOutput = {}));
var PutObjectLegalHoldRequest;
(function (PutObjectLegalHoldRequest) {
    PutObjectLegalHoldRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutObjectLegalHoldRequest.isa = (o) => smithy_client_1.isa(o, "PutObjectLegalHoldRequest");
})(PutObjectLegalHoldRequest = exports.PutObjectLegalHoldRequest || (exports.PutObjectLegalHoldRequest = {}));
var PutObjectLockConfigurationOutput;
(function (PutObjectLockConfigurationOutput) {
    PutObjectLockConfigurationOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutObjectLockConfigurationOutput.isa = (o) => smithy_client_1.isa(o, "PutObjectLockConfigurationOutput");
})(PutObjectLockConfigurationOutput = exports.PutObjectLockConfigurationOutput || (exports.PutObjectLockConfigurationOutput = {}));
var PutObjectLockConfigurationRequest;
(function (PutObjectLockConfigurationRequest) {
    PutObjectLockConfigurationRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutObjectLockConfigurationRequest.isa = (o) => smithy_client_1.isa(o, "PutObjectLockConfigurationRequest");
})(PutObjectLockConfigurationRequest = exports.PutObjectLockConfigurationRequest || (exports.PutObjectLockConfigurationRequest = {}));
var PutObjectOutput;
(function (PutObjectOutput) {
    PutObjectOutput.filterSensitiveLog = (obj) => (Object.assign(Object.assign(Object.assign({}, obj), (obj.SSEKMSKeyId && { SSEKMSKeyId: smithy_client_1.SENSITIVE_STRING })), (obj.SSEKMSEncryptionContext && { SSEKMSEncryptionContext: smithy_client_1.SENSITIVE_STRING })));
    PutObjectOutput.isa = (o) => smithy_client_1.isa(o, "PutObjectOutput");
})(PutObjectOutput = exports.PutObjectOutput || (exports.PutObjectOutput = {}));
var PutObjectRequest;
(function (PutObjectRequest) {
    PutObjectRequest.filterSensitiveLog = (obj) => (Object.assign(Object.assign(Object.assign(Object.assign({}, obj), (obj.SSEKMSEncryptionContext && { SSEKMSEncryptionContext: smithy_client_1.SENSITIVE_STRING })), (obj.SSECustomerKey && { SSECustomerKey: smithy_client_1.SENSITIVE_STRING })), (obj.SSEKMSKeyId && { SSEKMSKeyId: smithy_client_1.SENSITIVE_STRING })));
    PutObjectRequest.isa = (o) => smithy_client_1.isa(o, "PutObjectRequest");
})(PutObjectRequest = exports.PutObjectRequest || (exports.PutObjectRequest = {}));
var PutObjectRetentionOutput;
(function (PutObjectRetentionOutput) {
    PutObjectRetentionOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutObjectRetentionOutput.isa = (o) => smithy_client_1.isa(o, "PutObjectRetentionOutput");
})(PutObjectRetentionOutput = exports.PutObjectRetentionOutput || (exports.PutObjectRetentionOutput = {}));
var PutObjectRetentionRequest;
(function (PutObjectRetentionRequest) {
    PutObjectRetentionRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutObjectRetentionRequest.isa = (o) => smithy_client_1.isa(o, "PutObjectRetentionRequest");
})(PutObjectRetentionRequest = exports.PutObjectRetentionRequest || (exports.PutObjectRetentionRequest = {}));
var PutObjectTaggingOutput;
(function (PutObjectTaggingOutput) {
    PutObjectTaggingOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutObjectTaggingOutput.isa = (o) => smithy_client_1.isa(o, "PutObjectTaggingOutput");
})(PutObjectTaggingOutput = exports.PutObjectTaggingOutput || (exports.PutObjectTaggingOutput = {}));
var PutObjectTaggingRequest;
(function (PutObjectTaggingRequest) {
    PutObjectTaggingRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutObjectTaggingRequest.isa = (o) => smithy_client_1.isa(o, "PutObjectTaggingRequest");
})(PutObjectTaggingRequest = exports.PutObjectTaggingRequest || (exports.PutObjectTaggingRequest = {}));
var PutPublicAccessBlockRequest;
(function (PutPublicAccessBlockRequest) {
    PutPublicAccessBlockRequest.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    PutPublicAccessBlockRequest.isa = (o) => smithy_client_1.isa(o, "PutPublicAccessBlockRequest");
})(PutPublicAccessBlockRequest = exports.PutPublicAccessBlockRequest || (exports.PutPublicAccessBlockRequest = {}));
var QueueConfiguration;
(function (QueueConfiguration) {
    QueueConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    QueueConfiguration.isa = (o) => smithy_client_1.isa(o, "QueueConfiguration");
})(QueueConfiguration = exports.QueueConfiguration || (exports.QueueConfiguration = {}));
var QuoteFields;
(function (QuoteFields) {
    QuoteFields["ALWAYS"] = "ALWAYS";
    QuoteFields["ASNEEDED"] = "ASNEEDED";
})(QuoteFields = exports.QuoteFields || (exports.QuoteFields = {}));
var RecordsEvent;
(function (RecordsEvent) {
    RecordsEvent.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    RecordsEvent.isa = (o) => smithy_client_1.isa(o, "RecordsEvent");
})(RecordsEvent = exports.RecordsEvent || (exports.RecordsEvent = {}));
var Redirect;
(function (Redirect) {
    Redirect.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Redirect.isa = (o) => smithy_client_1.isa(o, "Redirect");
})(Redirect = exports.Redirect || (exports.Redirect = {}));
var RedirectAllRequestsTo;
(function (RedirectAllRequestsTo) {
    RedirectAllRequestsTo.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    RedirectAllRequestsTo.isa = (o) => smithy_client_1.isa(o, "RedirectAllRequestsTo");
})(RedirectAllRequestsTo = exports.RedirectAllRequestsTo || (exports.RedirectAllRequestsTo = {}));
var ReplicationConfiguration;
(function (ReplicationConfiguration) {
    ReplicationConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ReplicationConfiguration.isa = (o) => smithy_client_1.isa(o, "ReplicationConfiguration");
})(ReplicationConfiguration = exports.ReplicationConfiguration || (exports.ReplicationConfiguration = {}));
var ReplicationRule;
(function (ReplicationRule) {
    ReplicationRule.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ReplicationRule.isa = (o) => smithy_client_1.isa(o, "ReplicationRule");
})(ReplicationRule = exports.ReplicationRule || (exports.ReplicationRule = {}));
var ReplicationRuleAndOperator;
(function (ReplicationRuleAndOperator) {
    ReplicationRuleAndOperator.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ReplicationRuleAndOperator.isa = (o) => smithy_client_1.isa(o, "ReplicationRuleAndOperator");
})(ReplicationRuleAndOperator = exports.ReplicationRuleAndOperator || (exports.ReplicationRuleAndOperator = {}));
var ReplicationRuleFilter;
(function (ReplicationRuleFilter) {
    ReplicationRuleFilter.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ReplicationRuleFilter.isa = (o) => smithy_client_1.isa(o, "ReplicationRuleFilter");
})(ReplicationRuleFilter = exports.ReplicationRuleFilter || (exports.ReplicationRuleFilter = {}));
var ReplicationTime;
(function (ReplicationTime) {
    ReplicationTime.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ReplicationTime.isa = (o) => smithy_client_1.isa(o, "ReplicationTime");
})(ReplicationTime = exports.ReplicationTime || (exports.ReplicationTime = {}));
var ReplicationTimeValue;
(function (ReplicationTimeValue) {
    ReplicationTimeValue.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ReplicationTimeValue.isa = (o) => smithy_client_1.isa(o, "ReplicationTimeValue");
})(ReplicationTimeValue = exports.ReplicationTimeValue || (exports.ReplicationTimeValue = {}));
var RequestPaymentConfiguration;
(function (RequestPaymentConfiguration) {
    RequestPaymentConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    RequestPaymentConfiguration.isa = (o) => smithy_client_1.isa(o, "RequestPaymentConfiguration");
})(RequestPaymentConfiguration = exports.RequestPaymentConfiguration || (exports.RequestPaymentConfiguration = {}));
var RequestProgress;
(function (RequestProgress) {
    RequestProgress.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    RequestProgress.isa = (o) => smithy_client_1.isa(o, "RequestProgress");
})(RequestProgress = exports.RequestProgress || (exports.RequestProgress = {}));
var RestoreObjectOutput;
(function (RestoreObjectOutput) {
    RestoreObjectOutput.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    RestoreObjectOutput.isa = (o) => smithy_client_1.isa(o, "RestoreObjectOutput");
})(RestoreObjectOutput = exports.RestoreObjectOutput || (exports.RestoreObjectOutput = {}));
var RestoreObjectRequest;
(function (RestoreObjectRequest) {
    RestoreObjectRequest.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.RestoreRequest && { RestoreRequest: RestoreRequest.filterSensitiveLog(obj.RestoreRequest) })));
    RestoreObjectRequest.isa = (o) => smithy_client_1.isa(o, "RestoreObjectRequest");
})(RestoreObjectRequest = exports.RestoreObjectRequest || (exports.RestoreObjectRequest = {}));
var RestoreRequest;
(function (RestoreRequest) {
    RestoreRequest.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.OutputLocation && { OutputLocation: OutputLocation.filterSensitiveLog(obj.OutputLocation) })));
    RestoreRequest.isa = (o) => smithy_client_1.isa(o, "RestoreRequest");
})(RestoreRequest = exports.RestoreRequest || (exports.RestoreRequest = {}));
var RestoreRequestType;
(function (RestoreRequestType) {
    RestoreRequestType["SELECT"] = "SELECT";
})(RestoreRequestType = exports.RestoreRequestType || (exports.RestoreRequestType = {}));
var RoutingRule;
(function (RoutingRule) {
    RoutingRule.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    RoutingRule.isa = (o) => smithy_client_1.isa(o, "RoutingRule");
})(RoutingRule = exports.RoutingRule || (exports.RoutingRule = {}));
var S3KeyFilter;
(function (S3KeyFilter) {
    S3KeyFilter.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    S3KeyFilter.isa = (o) => smithy_client_1.isa(o, "S3KeyFilter");
})(S3KeyFilter = exports.S3KeyFilter || (exports.S3KeyFilter = {}));
var S3Location;
(function (S3Location) {
    S3Location.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.Encryption && { Encryption: Encryption.filterSensitiveLog(obj.Encryption) })));
    S3Location.isa = (o) => smithy_client_1.isa(o, "S3Location");
})(S3Location = exports.S3Location || (exports.S3Location = {}));
var ScanRange;
(function (ScanRange) {
    ScanRange.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    ScanRange.isa = (o) => smithy_client_1.isa(o, "ScanRange");
})(ScanRange = exports.ScanRange || (exports.ScanRange = {}));
var SelectObjectContentEventStream;
(function (SelectObjectContentEventStream) {
    SelectObjectContentEventStream.visit = (value, visitor) => {
        if (value.Stats !== undefined)
            return visitor.Stats(value.Stats);
        if (value.End !== undefined)
            return visitor.End(value.End);
        if (value.Progress !== undefined)
            return visitor.Progress(value.Progress);
        if (value.Records !== undefined)
            return visitor.Records(value.Records);
        if (value.Cont !== undefined)
            return visitor.Cont(value.Cont);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
    SelectObjectContentEventStream.filterSensitiveLog = (obj) => {
        if (obj.Stats !== undefined)
            return { Stats: StatsEvent.filterSensitiveLog(obj.Stats) };
        if (obj.End !== undefined)
            return { End: EndEvent.filterSensitiveLog(obj.End) };
        if (obj.Progress !== undefined)
            return { Progress: ProgressEvent.filterSensitiveLog(obj.Progress) };
        if (obj.Records !== undefined)
            return { Records: RecordsEvent.filterSensitiveLog(obj.Records) };
        if (obj.Cont !== undefined)
            return { Cont: ContinuationEvent.filterSensitiveLog(obj.Cont) };
        if (obj.$unknown !== undefined)
            return { [obj.$unknown[0]]: "UNKNOWN" };
    };
})(SelectObjectContentEventStream = exports.SelectObjectContentEventStream || (exports.SelectObjectContentEventStream = {}));
var SelectObjectContentOutput;
(function (SelectObjectContentOutput) {
    SelectObjectContentOutput.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.Payload && { Payload: "STREAMING_CONTENT" })));
    SelectObjectContentOutput.isa = (o) => smithy_client_1.isa(o, "SelectObjectContentOutput");
})(SelectObjectContentOutput = exports.SelectObjectContentOutput || (exports.SelectObjectContentOutput = {}));
var SelectObjectContentRequest;
(function (SelectObjectContentRequest) {
    SelectObjectContentRequest.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.SSECustomerKey && { SSECustomerKey: smithy_client_1.SENSITIVE_STRING })));
    SelectObjectContentRequest.isa = (o) => smithy_client_1.isa(o, "SelectObjectContentRequest");
})(SelectObjectContentRequest = exports.SelectObjectContentRequest || (exports.SelectObjectContentRequest = {}));
var SelectParameters;
(function (SelectParameters) {
    SelectParameters.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    SelectParameters.isa = (o) => smithy_client_1.isa(o, "SelectParameters");
})(SelectParameters = exports.SelectParameters || (exports.SelectParameters = {}));
var ServerSideEncryptionByDefault;
(function (ServerSideEncryptionByDefault) {
    ServerSideEncryptionByDefault.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.KMSMasterKeyID && { KMSMasterKeyID: smithy_client_1.SENSITIVE_STRING })));
    ServerSideEncryptionByDefault.isa = (o) => smithy_client_1.isa(o, "ServerSideEncryptionByDefault");
})(ServerSideEncryptionByDefault = exports.ServerSideEncryptionByDefault || (exports.ServerSideEncryptionByDefault = {}));
var ServerSideEncryptionConfiguration;
(function (ServerSideEncryptionConfiguration) {
    ServerSideEncryptionConfiguration.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.Rules && { Rules: obj.Rules.map((item) => ServerSideEncryptionRule.filterSensitiveLog(item)) })));
    ServerSideEncryptionConfiguration.isa = (o) => smithy_client_1.isa(o, "ServerSideEncryptionConfiguration");
})(ServerSideEncryptionConfiguration = exports.ServerSideEncryptionConfiguration || (exports.ServerSideEncryptionConfiguration = {}));
var ServerSideEncryptionRule;
(function (ServerSideEncryptionRule) {
    ServerSideEncryptionRule.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.ApplyServerSideEncryptionByDefault && {
        ApplyServerSideEncryptionByDefault: ServerSideEncryptionByDefault.filterSensitiveLog(obj.ApplyServerSideEncryptionByDefault),
    })));
    ServerSideEncryptionRule.isa = (o) => smithy_client_1.isa(o, "ServerSideEncryptionRule");
})(ServerSideEncryptionRule = exports.ServerSideEncryptionRule || (exports.ServerSideEncryptionRule = {}));
var SourceSelectionCriteria;
(function (SourceSelectionCriteria) {
    SourceSelectionCriteria.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    SourceSelectionCriteria.isa = (o) => smithy_client_1.isa(o, "SourceSelectionCriteria");
})(SourceSelectionCriteria = exports.SourceSelectionCriteria || (exports.SourceSelectionCriteria = {}));
var SSEKMS;
(function (SSEKMS) {
    SSEKMS.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.KeyId && { KeyId: smithy_client_1.SENSITIVE_STRING })));
    SSEKMS.isa = (o) => smithy_client_1.isa(o, "SSEKMS");
})(SSEKMS = exports.SSEKMS || (exports.SSEKMS = {}));
var SseKmsEncryptedObjects;
(function (SseKmsEncryptedObjects) {
    SseKmsEncryptedObjects.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    SseKmsEncryptedObjects.isa = (o) => smithy_client_1.isa(o, "SseKmsEncryptedObjects");
})(SseKmsEncryptedObjects = exports.SseKmsEncryptedObjects || (exports.SseKmsEncryptedObjects = {}));
var SSES3;
(function (SSES3) {
    SSES3.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    SSES3.isa = (o) => smithy_client_1.isa(o, "SSES3");
})(SSES3 = exports.SSES3 || (exports.SSES3 = {}));
var Stats;
(function (Stats) {
    Stats.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Stats.isa = (o) => smithy_client_1.isa(o, "Stats");
})(Stats = exports.Stats || (exports.Stats = {}));
var StatsEvent;
(function (StatsEvent) {
    StatsEvent.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    StatsEvent.isa = (o) => smithy_client_1.isa(o, "StatsEvent");
})(StatsEvent = exports.StatsEvent || (exports.StatsEvent = {}));
var StorageClassAnalysis;
(function (StorageClassAnalysis) {
    StorageClassAnalysis.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    StorageClassAnalysis.isa = (o) => smithy_client_1.isa(o, "StorageClassAnalysis");
})(StorageClassAnalysis = exports.StorageClassAnalysis || (exports.StorageClassAnalysis = {}));
var StorageClassAnalysisDataExport;
(function (StorageClassAnalysisDataExport) {
    StorageClassAnalysisDataExport.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    StorageClassAnalysisDataExport.isa = (o) => smithy_client_1.isa(o, "StorageClassAnalysisDataExport");
})(StorageClassAnalysisDataExport = exports.StorageClassAnalysisDataExport || (exports.StorageClassAnalysisDataExport = {}));
var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Tag.isa = (o) => smithy_client_1.isa(o, "Tag");
})(Tag = exports.Tag || (exports.Tag = {}));
var Tagging;
(function (Tagging) {
    Tagging.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Tagging.isa = (o) => smithy_client_1.isa(o, "Tagging");
})(Tagging = exports.Tagging || (exports.Tagging = {}));
var TargetGrant;
(function (TargetGrant) {
    TargetGrant.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    TargetGrant.isa = (o) => smithy_client_1.isa(o, "TargetGrant");
})(TargetGrant = exports.TargetGrant || (exports.TargetGrant = {}));
var TopicConfiguration;
(function (TopicConfiguration) {
    TopicConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    TopicConfiguration.isa = (o) => smithy_client_1.isa(o, "TopicConfiguration");
})(TopicConfiguration = exports.TopicConfiguration || (exports.TopicConfiguration = {}));
var Transition;
(function (Transition) {
    Transition.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    Transition.isa = (o) => smithy_client_1.isa(o, "Transition");
})(Transition = exports.Transition || (exports.Transition = {}));
var UploadPartCopyOutput;
(function (UploadPartCopyOutput) {
    UploadPartCopyOutput.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.SSEKMSKeyId && { SSEKMSKeyId: smithy_client_1.SENSITIVE_STRING })));
    UploadPartCopyOutput.isa = (o) => smithy_client_1.isa(o, "UploadPartCopyOutput");
})(UploadPartCopyOutput = exports.UploadPartCopyOutput || (exports.UploadPartCopyOutput = {}));
var UploadPartCopyRequest;
(function (UploadPartCopyRequest) {
    UploadPartCopyRequest.filterSensitiveLog = (obj) => (Object.assign(Object.assign(Object.assign({}, obj), (obj.CopySourceSSECustomerKey && { CopySourceSSECustomerKey: smithy_client_1.SENSITIVE_STRING })), (obj.SSECustomerKey && { SSECustomerKey: smithy_client_1.SENSITIVE_STRING })));
    UploadPartCopyRequest.isa = (o) => smithy_client_1.isa(o, "UploadPartCopyRequest");
})(UploadPartCopyRequest = exports.UploadPartCopyRequest || (exports.UploadPartCopyRequest = {}));
var UploadPartOutput;
(function (UploadPartOutput) {
    UploadPartOutput.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.SSEKMSKeyId && { SSEKMSKeyId: smithy_client_1.SENSITIVE_STRING })));
    UploadPartOutput.isa = (o) => smithy_client_1.isa(o, "UploadPartOutput");
})(UploadPartOutput = exports.UploadPartOutput || (exports.UploadPartOutput = {}));
var UploadPartRequest;
(function (UploadPartRequest) {
    UploadPartRequest.filterSensitiveLog = (obj) => (Object.assign(Object.assign({}, obj), (obj.SSECustomerKey && { SSECustomerKey: smithy_client_1.SENSITIVE_STRING })));
    UploadPartRequest.isa = (o) => smithy_client_1.isa(o, "UploadPartRequest");
})(UploadPartRequest = exports.UploadPartRequest || (exports.UploadPartRequest = {}));
var VersioningConfiguration;
(function (VersioningConfiguration) {
    VersioningConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    VersioningConfiguration.isa = (o) => smithy_client_1.isa(o, "VersioningConfiguration");
})(VersioningConfiguration = exports.VersioningConfiguration || (exports.VersioningConfiguration = {}));
var WebsiteConfiguration;
(function (WebsiteConfiguration) {
    WebsiteConfiguration.filterSensitiveLog = (obj) => (Object.assign({}, obj));
    WebsiteConfiguration.isa = (o) => smithy_client_1.isa(o, "WebsiteConfiguration");
})(WebsiteConfiguration = exports.WebsiteConfiguration || (exports.WebsiteConfiguration = {}));
