// @aws-sdk/middleware-apply-body-checksum/dist/cjs/applyMd5BodyChecksumMiddleware.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getApplyMd5BodyChecksumPlugin = exports.applyMd5BodyChecksumMiddlewareOptions = exports.applyMd5BodyChecksumMiddleware = void 0;
var tslib_1 = require("tslib");
var is_array_buffer_1 = require("@aws-sdk/is-array-buffer");
var protocol_http_1 = require("@aws-sdk/protocol-http");
function applyMd5BodyChecksumMiddleware(options) {
    var _this = this;
    return function (next) { return function (args) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
        var request, body, headers, digest, hash, _a, _b, _c, _d, _e;
        var _f, _g;
        return tslib_1.__generator(this, function (_h) {
            switch (_h.label) {
                case 0:
                    request = args.request;
                    if (!protocol_http_1.HttpRequest.isInstance(request)) return [3 /*break*/, 2];
                    body = request.body, headers = request.headers;
                    if (!!hasHeader("Content-MD5", headers)) return [3 /*break*/, 2];
                    digest = void 0;
                    if (body === undefined || typeof body === "string" || ArrayBuffer.isView(body) || is_array_buffer_1.isArrayBuffer(body)) {
                        hash = new options.md5();
                        hash.update(body || "");
                        digest = hash.digest();
                    }
                    else {
                        digest = options.streamHasher(options.md5, body);
                    }
                    _a = [tslib_1.__assign({}, request)];
                    _f = {};
                    _b = [tslib_1.__assign({}, headers)];
                    _g = {};
                    _c = "Content-MD5";
                    _e = (_d = options).base64Encoder;
                    return [4 /*yield*/, digest];
                case 1:
                    request = tslib_1.__assign.apply(void 0, _a.concat([(_f.headers = tslib_1.__assign.apply(void 0, _b.concat([(_g[_c] = _e.apply(_d, [_h.sent()]), _g)])), _f)]));
                    _h.label = 2;
                case 2: return [2 /*return*/, next(tslib_1.__assign(tslib_1.__assign({}, args), { request: request }))];
            }
        });
    }); }; };
}
exports.applyMd5BodyChecksumMiddleware = applyMd5BodyChecksumMiddleware;
exports.applyMd5BodyChecksumMiddlewareOptions = {
    name: "applyMd5BodyChecksumMiddleware",
    step: "build",
    tags: ["SET_CONTENT_MD5", "BODY_CHECKSUM"],
};
exports.getApplyMd5BodyChecksumPlugin = function (config) { return ({
    applyToStack: function (clientStack) {
        clientStack.add(applyMd5BodyChecksumMiddleware(config), exports.applyMd5BodyChecksumMiddlewareOptions);
    },
}); };
function hasHeader(soughtHeader, headers) {
    var e_1, _a;
    soughtHeader = soughtHeader.toLowerCase();
    try {
        for (var _b = tslib_1.__values(Object.keys(headers)), _c = _b.next(); !_c.done; _c = _b.next()) {
            var headerName = _c.value;
            if (soughtHeader === headerName.toLowerCase()) {
                return true;
            }
        }
    }
    catch (e_1_1) { e_1 = { error: e_1_1 }; }
    finally {
        try {
            if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
        }
        finally { if (e_1) throw e_1.error; }
    }
    return false;
}
